--- civicrm-4.7.11-drupal/templates/CRM/common/civicrm.settings.php.template	2016-09-08 05:56:20.000000000 +1000
+++ civicrm-4.7.12-drupal/templates/CRM/common/civicrm.settings.php.template	2016-10-05 16:20:36.000000000 +1100
@@ -103,6 +103,14 @@
 }
 
 /**
+ * CiviCRM DSN Mode
+ * Used to determine if you want CiviCRM to automatically change the dsn to mysqli if its avaliable.
+ * Uncomment and edit below as necessary
+ */
+// define ('DB_DSN_MODE', 'auto');
+
+
+/**
  * CiviCRM Logging Database
  *
  * Used to point to a different database to use for logging (if desired). If unset defaults to equal CIVICRM_DSN.
@@ -212,10 +220,10 @@
  //  global $civicrm_setting;
 
  // Override the Temporary Files directory.
- // $civicrm_setting['Directory Preferences']['customFileUploadDir'] = '/path/to/upload';
+ // $civicrm_setting['Directory Preferences']['uploadDir'] = '/path/to/upload-dir' ;
 
  // Override the custom files upload directory.
- // $civicrm_setting['Directory Preferences']['uploadDir'] = '/path/to/upload-dir' ;
+ // $civicrm_setting['Directory Preferences']['customFileUploadDir'] = '/path/to/custom-dir';
 
  // Override the images directory.
  // $civicrm_setting['Directory Preferences']['imageUploadDir'] = '/path/to/image-upload-dir' ;
