--- civicrm-5.40.4-drupal/templates/CRM/common/civicrm.settings.php.template	2021-08-07 00:35:31.000000000 +1000
+++ civicrm-5.47.0-drupal/templates/CRM/common/civicrm.settings.php.template	2022-02-06 13:37:22.000000000 +1100
@@ -176,5 +176,3 @@
  *
  */
-
-global $civicrm_root;
 
@@ -195,7 +193,19 @@
  *
  */
 //if (!defined('CIVICRM_TEMPLATE_COMPILE_CHECK')) {
-//  define( 'CIVICRM_TEMPLATE_COMPILE_CHECK', FALSE);
+//  define('CIVICRM_TEMPLATE_COMPILE_CHECK', FALSE);
+//}
+
+/**
+ * Smarty escape on output.
+ *
+ * This tells smarty to pass all variables through the escape function
+ * unless they are piped to smarty:nodefaults (eg. {$myScript|smarty:nodefaults}
+ * At this stage it should only be enabled on development sites.
+ * @see https://github.com/civicrm/civicrm-core/pull/21935
+ */
+//if (!defined('CIVICRM_SMARTY_DEFAULT_ESCAPE')) {
+//  define('CIVICRM_SMARTY_DEFAULT_ESCAPE', TRUE);
 //}
 
 /**
