--- civicrm_entity.module.orig  2020-01-29 14:45:57.292712358 +1100
+++ civicrm_entity.module       2020-01-29 14:40:15.049485309 +1100
@@ -2622,7 +2622,7 @@
   civicrm_initialize();
   try {
     $result = civicrm_api3($civicrm_entity, 'get', array(
-      'options' => array('limit' => 100),
+      'options' => array('limit' => 1000),
     ));
   } catch (CiviCRM_API3_Exception $e) {
     return array('');
