--- civicrm-4.5.8-drupal/templates/CRM/common/civicrm.settings.php.template	2015-03-06 14:16:39.000000000 +1100
+++ civicrm-4.6.0-drupal/templates/CRM/common/civicrm.settings.php.template	2015-04-04 01:31:36.000000000 +1100
@@ -1,9 +1,9 @@
 <?php
 /*
  +--------------------------------------------------------------------+
- | CiviCRM version 4.4                                                |
+ | CiviCRM version 4.6                                                |
  +--------------------------------------------------------------------+
- | Copyright CiviCRM LLC (c) 2004-2013                                |
+ | Copyright CiviCRM LLC (c) 2004-2015                                |
  +--------------------------------------------------------------------+
  | This file is a part of CiviCRM.                                    |
  |                                                                    |
