--- civicrm-4.7.7-drupal/templates/CRM/common/civicrm.settings.php.template	2016-05-05 06:24:45.000000000 +1000
+++ civicrm-4.7.11-drupal/templates/CRM/common/civicrm.settings.php.template	2016-09-08 05:56:20.000000000 +1000
@@ -253,6 +253,13 @@
  // Override the CMS root path defined by cmsRootPath.
  // define('CIVICRM_CMSDIR', '/path/to/install/root/');
 
+ // Opt-out of announcements by the CiviCRM core team for releases, paid services, events, etc. Separate each preferred optout category with a comma:
+ //   "offers": paid service offers
+ //   "asks": requests for donations or membership signup/renewal to CiviCRM
+ //   "releases": major release announcements
+ //   "events": announcements of local/national upcoming events
+ // $civicrm_setting['CiviCRM Preferences']['communityMessagesUrl'] = 'https://alert.civicrm.org/alert?prot=1&ver={ver}&uf={uf}&sid={sid}&lang={lang}&co={co}&optout=offers,asks';
+
 
 /**
  * If you are using any CiviCRM script in the bin directory that
