--- civicrm-5.3.2-drupal/templates/CRM/common/civicrm.settings.php.template	2018-07-26 16:17:18.000000000 +1000
+++ civicrm-5.10.4-drupal/templates/CRM/common/civicrm.settings.php.template	2019-02-23 15:52:12.000000000 +1100
@@ -368,7 +368,12 @@
  * The standard port for Memcache & APCCache is 11211. For Redis it is 6379.
  */
 if (!defined('CIVICRM_DB_CACHE_PORT')) {
-  define('CIVICRM_DB_CACHE_PORT', 11211 );
+  if (CIVICRM_DB_CACHE_CLASS === 'Redis') {
+    define('CIVICRM_DB_CACHE_PORT', 6379 );
+  }
+  else {
+    define('CIVICRM_DB_CACHE_PORT', 11211);
+  }
 }
 
 /**
@@ -400,6 +405,14 @@
 }
 
 /**
+ * The cache system traditionally allowed a wide range of cache-keys, but some
+ * cache-keys are prohibited by PSR-16.
+ */
+if (!defined('CIVICRM_PSR16_STRICT')) {
+  define('CIVICRM_PSR16_STRICT', FALSE);
+}
+
+/**
  * If you have multilingual site and you are using the "inherit CMS language"
  * configuration option, but wish to, for example, use fr_CA instead of the
  * default fr_FR (for French), set one or more of the constants below to an
