--- civicrm-4.7.6-drupal/templates/CRM/common/civicrm.settings.php.template	2016-04-08 06:29:24.000000000 +1000
+++ civicrm-4.7.7-drupal/templates/CRM/common/civicrm.settings.php.template	2016-05-05 06:24:45.000000000 +1000
@@ -247,6 +247,9 @@
  // Disable automatic download / installation of extensions
  // $civicrm_setting['Extension Preferences']['ext_repo_url'] = false;
 
+ // set triggers to be managed offline per CRM-18212
+ // $civicrm_setting['CiviCRM Preferences']['logging_no_trigger_permission'] = 1;
+
  // Override the CMS root path defined by cmsRootPath.
  // define('CIVICRM_CMSDIR', '/path/to/install/root/');
 
@@ -322,14 +325,16 @@
 }
 
 /**
- * Change this if you are not using the standard port for your cache server
+ * Change this if you are not using the standard port for your cache server.
+ *
+ * The standard port for Memcache & APCCache is 11211. For Redis it is 6379.
  */
 if (!defined('CIVICRM_DB_CACHE_PORT')) {
   define('CIVICRM_DB_CACHE_PORT', 11211 );
 }
 
 /**
- * Change this if your cache server requires a password (currently only works 
+ * Change this if your cache server requires a password (currently only works
  * with Redis)
  */
 if (!defined('CIVICRM_DB_CACHE_PASSWORD')) {
