--- /tmp/crm.ckeditor.js	2016-04-28 11:06:55.635855491 +1000
+++ ../civicrm/js/wysiwyg/crm.ckeditor.js	2016-04-28 11:13:03.171839105 +1000
@@ -78,7 +78,8 @@
         filebrowserImageUploadUrl : '',
         filebrowserFlashUploadUrl : '',
         bodyClass                 : 'cke_editable',
-    	contentsCss               : '/sites/all/themes/cbootf/css/editor.css',
+        contentsCss               : '/sites/all/themes/cbootf/css/editor.css',
+        stylesSet                 : 'civicrm:/sites/all/themes/cbootf/js/ckeditor.styles.js',
         scayt_autoStartup         : true,
         scayt_sLang               : 'en_GB',
         forcePasteAsPlainText     : true,
