--- a/vendor/dompdf/dompdf/lib/html5lib/TreeBuilder.php	2015-12-07 15:07:13.000000000 +1100
+++ b/vendor/dompdf/dompdf/lib/html5lib/TreeBuilder.php	2015-12-07 15:07:13.000000000 +1100
@@ -3365,7 +3365,7 @@
     }
 
     private function getElementCategory($node) {
-        if (!is_object($node)) debug_print_backtrace();
+        if (!is_object($node)) CRM_Core_Error::backtrace("backTrace", TRUE);
         $name = $node->tagName;
         if(in_array($name, $this->special))
             return self::SPECIAL;
