From a80769d49eccb9ff257a92fdc60027d8bdec138c Mon Sep 17 00:00:00 2001
From: Seamus Lee <seamuslee001@gmail.com>
Date: Thu, 18 May 2017 08:04:32 +1000
Subject: [PATCH] CRM-20487 have pingback url use https

---
 CRM/Utils/VersionCheck.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CRM/Utils/VersionCheck.php b/CRM/Utils/VersionCheck.php
index 6cd1507bd69..4dea0603ada 100644
--- a/CRM/Utils/VersionCheck.php
+++ b/CRM/Utils/VersionCheck.php
@@ -70,7 +70,7 @@ class CRM_Utils_VersionCheck {
   /**
    * @var string
    */
-  public $pingbackUrl = 'http://latest.civicrm.org/stable.php?format=json';
+  public $pingbackUrl = 'https://latest.civicrm.org/stable.php?format=json';
 
   /**
    * Pingback params
