diff --git a/extern/url.php b/extern/url.php
index 0819f9c..4c89886 100644
--- a/extern/url.php
+++ b/extern/url.php
@@ -54,6 +54,9 @@
   }
 }
 
+// CRM-18320 - Fix encoded ampersands (see CRM_Utils_System::redirect)
+$url = str_replace('&amp;', '&', $url);
+
 // CRM-17953 - The CMS is not bootstrapped so cannot use CRM_Utils_System::redirect
 header('Location: ' . $url);
 CRM_Utils_System::civiExit();
