--- CRM/Export/BAO/Export.php	2014-03-14 22:56:22.075270582 +1100
+++ civicrm-4.4.4-drupal/CRM/Export/BAO/Export.php	2014-02-07 14:29:30.000000000 +1100
@@ -549,6 +549,20 @@
       }
     }
 
+    // CRM-13982 - check if is deleted
+    $excludeTrashed = TRUE;
+    foreach ($params as $value) {
+      if ($value[0] == 'contact_is_deleted') {
+        $excludeTrashed = FALSE;
+      }
+    }
+    if (empty($where) && $excludeTrashed) {
+      $where = "WHERE contact_a.is_deleted != 1";
+    }
+    elseif ($excludeTrashed) {
+      $where .= " AND contact_a.is_deleted != 1";
+    }
+
     $queryString = "$select $from $where $having";
 
     $groupBy = "";
