diff --git a/includes/context.inc b/includes/context.inc
index e933bec..36b3620 100644
--- a/includes/context.inc
+++ b/includes/context.inc
@@ -1424,6 +1424,9 @@ function ctools_context_get_context_from_relationships($relationships, &$context
     elseif (!empty($contexts[$rdata['context']])) {
       $rcontexts = $contexts[$rdata['context']];
     }
+    else {
+      $rcontexts = null;
+    }
 
     $cid = ctools_context_id($rdata, 'relationship');
     if ($context = ctools_context_get_context_from_relationship($rdata, $rcontexts)) {
