diff --git a/entityreference.module b/entityreference.module
index fab22da..0a7c120 100644
--- a/entityreference.module
+++ b/entityreference.module
@@ -1126,13 +1126,13 @@ function entityreference_autocomplete_callback_get_matches($type, $field, $insta
     $entity_labels = $handler->getReferencableEntities($tag_last, $instance['widget']['settings']['match_operator'], 10);
     $denied_label = t(ENTITYREFERENCE_DENIED);
     // Loop through the products and convert them into autocomplete output.
-    foreach ($entity_labels as $values) {
+    foreach ($entity_labels as $contentType => $values) {
       foreach ($values as $entity_id => $label) {
         // Never autocomplete entities that aren't accessible.
         if ($label == $denied_label) {
           continue;
         }
-        $key = "$label ($entity_id)";
+        $key = "$label ($entity_id) [$contentType]";
         if (isset($instance['widget']['settings']['hide_ids'])) {
           $key = $instance['widget']['settings']['hide_ids'] ? $label : "$label ($entity_id)";
         }
@@ -1142,7 +1142,7 @@ function entityreference_autocomplete_callback_get_matches($type, $field, $insta
         if (strpos($key, ',') !== FALSE || strpos($key, '"') !== FALSE) {
           $key = '"' . str_replace('"', '""', $key) . '"';
         }
-        $matches[$prefix . $key] = '<div class="reference-autocomplete">' . $label . '</div>';
+        $matches[$prefix . $key] = '<div class="reference-autocomplete">' . $label . ' ['. $contentType . ', '. $entity_id . ']' . '</div>';
       }
     }
   }
