diff --git a/fb_instant_articles.module b/fb_instant_articles.module
index 3c5e6f9..503d79b 100644
--- a/fb_instant_articles.module
+++ b/fb_instant_articles.module
@@ -162,26 +162,30 @@ function fb_instant_articles_html_head_alter(&$head_elements) {
  *   @link https://github.com/facebook/facebook-instant-articles-sdk-php/issues/14 this SDK issue @endlink is in resolved.
  */
 function fb_instant_articles_init() {
-  $enable_transformer_logging = variable_get('fb_instant_articles_enable_logging');
-  $appender = array(
-    'class' => $enable_transformer_logging ? '\Drupal\fb_instant_articles\DrupalLoggerAppender' : 'LoggerAppenderNull',
-    'layout' => array(
-      'class' => 'LoggerLayoutSimple'
-    ),
-  );
-  $configuration = array(
-    'rootLogger' => array(
+  // Ensure the Logger class exists before using it. This can occur whilst
+  // moving sites and rebuild composer dependencies.
+  if (class_exists('\Logger')) {
+    $enable_transformer_logging = variable_get('fb_instant_articles_enable_logging');
+    $appender = array(
+      'class' => $enable_transformer_logging ? '\Drupal\fb_instant_articles\DrupalLoggerAppender' : 'LoggerAppenderNull',
+      'layout' => array(
+        'class' => 'LoggerLayoutSimple'
+      ),
+    );
+    $configuration = array(
+      'rootLogger' => array(
+        'appenders' => array(
+          'facebook-instantarticles-transformer',
+          'facebook-instantarticles-client',
+        ),
+      ),
       'appenders' => array(
-        'facebook-instantarticles-transformer',
-        'facebook-instantarticles-client',
+        'facebook-instantarticles-transformer' => $appender,
+        'facebook-instantarticles-client' => $appender,
       ),
-    ),
-    'appenders' => array(
-      'facebook-instantarticles-transformer' => $appender,
-      'facebook-instantarticles-client' => $appender,
-    ),
-  );
-  \Logger::configure($configuration);
+    );
+    \Logger::configure($configuration);
+  }
 }
 
 /**
