diff --git a/lib/Drupal/gmap/GmapDefaults.php b/lib/Drupal/gmap/GmapDefaults.php
index 4b36fc8..2fb12ac 100644
--- a/lib/Drupal/gmap/GmapDefaults.php
+++ b/lib/Drupal/gmap/GmapDefaults.php
@@ -104,7 +104,7 @@ class GmapDefaults {
       'v' => variable_get('gmap_api_version', GMAP_API_VERSION),
       'language' => $langcode,
       'sensor' => 'false',
-      'libraries' => implode(array_merge(variable_get('gmap_api_libraries', array()), gmap_module_invoke('libraries', $m)), ','),
+      'libraries' => implode(',', array_merge(variable_get('gmap_api_libraries', array()), gmap_module_invoke('libraries', $m))),
     );
     if ($key = gmap_get_key()) {
       $query['key'] = $key;
