diff -ur ./ang/crmMailing/EditMailingCtrl/2step.html /var/www/civicrm/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html
--- ./ang/crmMailing/EditMailingCtrl/2step.html	2020-08-06 12:54:52.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html	2020-09-01 14:45:49.314548537 +1000
@@ -14,9 +14,9 @@
             </div>
             <span ng-model="placeholder" crm-ui-validate="mailing.body_html || mailing.body_text"></span>
           </div>
-          <div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
+          <!-- div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
             <div crm-attachments="attachments"></div>
-          </div>
+          </div -->
           <div crm-ui-tab id="tab-header" crm-title="ts('Header and Footer')" ng-if="mailingFields.header_id.options.length > 0 || mailingFields.footer_id.options.length > 0">
             <div crm-mailing-block-header-footer crm-mailing="mailing"></div>
           </div>
diff -ur ./api/v3/Generic/Setvalue.php /var/www/civicrm/sites/all/modules/civicrm/api/v3/Generic/Setvalue.php
--- ./api/v3/Generic/Setvalue.php	2020-01-02 13:36:08.000000000 +1100
+++ /var/www/civicrm/sites/all/modules/civicrm/api/v3/Generic/Setvalue.php	2020-08-31 21:11:02.329018837 +1000
@@ -89,6 +89,7 @@
       break;
 
     case CRM_Utils_Type::T_DATE:
+    case CRM_Utils_Type::T_DATE + CRM_Utils_Type::T_TIME:
       $value = CRM_Utils_Type::escape($value, "Date", FALSE);
       if (!$value) {
         return civicrm_api3_create_error("Param '$field' is not a date. format YYYYMMDD or YYYYMMDDHHMMSS");
diff -ur ./api/v3/Generic.php /var/www/civicrm/sites/all/modules/civicrm/api/v3/Generic.php
--- ./api/v3/Generic.php	2020-08-06 12:54:52.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/api/v3/Generic.php	2020-09-01 14:52:23.831090401 +1000
@@ -308,7 +308,7 @@
     return $result['values'][0];
   }
   if ($result['count'] !== 1) {
-    return civicrm_api3_create_error("Expected one " . $apiRequest['entity'] . " but found " . $result['count'], ['count' => $result['count']]);
+    return civicrm_api3_create_error("Get single expected one " . $apiRequest['entity'] . " but found " . $result['count'], ['count' => $result['count']]);
   }
   return civicrm_api3_create_error("Undefined behavior");
 }
@@ -329,7 +329,7 @@
     return $result;
   }
   if ($result['count'] !== 1) {
-    $result = civicrm_api3_create_error("Expected one " . $apiRequest['entity'] . " but found " . $result['count'], ['count' => $result['count']]);
+    $result = civicrm_api3_create_error("Get value expected one " . $apiRequest['entity'] . " but found " . $result['count'], ['count' => $result['count']]);
     return $result;
   }
 
diff -ur ./bower_components/ckeditor/plugins/table/dialogs/table.js /var/www/civicrm/sites/all/modules/civicrm/bower_components/ckeditor/plugins/table/dialogs/table.js
--- ./bower_components/ckeditor/plugins/table/dialogs/table.js	2020-03-05 00:19:40.000000000 +1100
+++ /var/www/civicrm/sites/all/modules/civicrm/bower_components/ckeditor/plugins/table/dialogs/table.js	2020-09-01 14:40:41.911037705 +1000
@@ -11,12 +11,11 @@
 b.setStyle("height",c.txtHeight):b.removeStyle("height");c.txtWidth?b.setStyle("width",c.txtWidth):b.removeStyle("width");b.getAttribute("style")||b.removeAttribute("style")}if(this._.selectedElement)try{d.selectBookmarks(e)}catch(n){}else a.insertElement(b),setTimeout(function(){var d=new CKEDITOR.dom.element(b.$.rows[0].cells[0]),c=a.createRange();c.moveToPosition(d,CKEDITOR.POSITION_AFTER_START);c.select()},0)},contents:[{id:"info",label:a.lang.table.title,elements:[{type:"hbox",widths:[null,null],
 styles:["vertical-align:top"],children:[{type:"vbox",padding:0,children:[{type:"text",id:"txtRows","default":3,label:a.lang.table.rows,required:!0,controlStyle:"width:5em",validate:t(a.lang.table.invalidRows),setup:function(d){this.setValue(d.$.rows.length)},commit:n},{type:"text",id:"txtCols","default":2,label:a.lang.table.columns,required:!0,controlStyle:"width:5em",validate:t(a.lang.table.invalidCols),setup:function(d){this.setValue(w(d))},commit:n},{type:"html",html:"\x26nbsp;"},{type:"select",
 id:"selHeaders",requiredContent:"th","default":"",label:a.lang.table.headers,items:[[a.lang.table.headersNone,""],[a.lang.table.headersRow,"row"],[a.lang.table.headersColumn,"col"],[a.lang.table.headersBoth,"both"]],setup:function(d){var a=this.getDialog();a.hasColumnHeaders=!0;for(var b=0;b<d.$.rows.length;b++){var c=d.$.rows[b].cells[0];if(c&&"th"!=c.nodeName.toLowerCase()){a.hasColumnHeaders=!1;break}}null!==d.$.tHead?this.setValue(a.hasColumnHeaders?"both":"row"):this.setValue(a.hasColumnHeaders?
-"col":"")},commit:n},{type:"text",id:"txtBorder",requiredContent:"table[border]","default":a.filter.check("table[border]")?1:0,label:a.lang.table.border,controlStyle:"width:3em",validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidBorder),setup:function(d){this.setValue(d.getAttribute("border")||"")},commit:function(d,a){this.getValue()?a.setAttribute("border",this.getValue()):a.removeAttribute("border")}},{id:"cmbAlign",type:"select",requiredContent:"table[align]","default":"",label:a.lang.common.align,
+"col":"")},commit:n},{type:"text",id:"txtBorder",requiredContent:"table[border]","default":0,label:a.lang.table.border,controlStyle:"width:3em",validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidBorder),setup:function(d){this.setValue(d.getAttribute("border")||"")},commit:function(d,a){this.getValue()?a.setAttribute("border",this.getValue()):a.removeAttribute("border")}},{id:"cmbAlign",type:"select",requiredContent:"table[align]","default":"",label:a.lang.common.align,
 items:[[a.lang.common.notSet,""],[a.lang.common.left,"left"],[a.lang.common.center,"center"],[a.lang.common.right,"right"]],setup:function(a){this.setValue(a.getAttribute("align")||"")},commit:function(a,e){this.getValue()?e.setAttribute("align",this.getValue()):e.removeAttribute("align")}}]},{type:"vbox",padding:0,children:[{type:"hbox",widths:["5em"],children:[{type:"text",id:"txtWidth",requiredContent:"table{width}",controlStyle:"width:5em",label:a.lang.common.width,title:a.lang.common.cssLengthTooltip,
-"default":a.filter.check("table{width}")?500>r.getSize("width")?"100%":500:0,getValue:v,validate:CKEDITOR.dialog.validate.cssLength(a.lang.common.invalidCssLength.replace("%1",a.lang.common.width)),onChange:function(){var a=this.getDialog().getContentElement("advanced","advStyles");a&&a.updateStyle("width",this.getValue())},setup:function(a){a=a.getStyle("width");this.setValue(a)},commit:n}]},{type:"hbox",widths:["5em"],children:[{type:"text",id:"txtHeight",requiredContent:"table{height}",controlStyle:"width:5em",
+"default":"100%",getValue:v,validate:CKEDITOR.dialog.validate.cssLength(a.lang.common.invalidCssLength.replace("%1",a.lang.common.width)),onChange:function(){var a=this.getDialog().getContentElement("advanced","advStyles");a&&a.updateStyle("width",this.getValue())},setup:function(a){a=a.getStyle("width");this.setValue(a)},commit:n}]},{type:"hbox",widths:["5em"],children:[{type:"text",id:"txtHeight",requiredContent:"table{height}",controlStyle:"width:5em",
 label:a.lang.common.height,title:a.lang.common.cssLengthTooltip,"default":"",getValue:v,validate:CKEDITOR.dialog.validate.cssLength(a.lang.common.invalidCssLength.replace("%1",a.lang.common.height)),onChange:function(){var a=this.getDialog().getContentElement("advanced","advStyles");a&&a.updateStyle("height",this.getValue())},setup:function(a){(a=a.getStyle("height"))&&this.setValue(a)},commit:n}]},{type:"html",html:"\x26nbsp;"},{type:"text",id:"txtCellSpace",requiredContent:"table[cellspacing]",
-controlStyle:"width:3em",label:a.lang.table.cellSpace,"default":a.filter.check("table[cellspacing]")?1:0,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellSpacing),setup:function(a){this.setValue(a.getAttribute("cellSpacing")||"")},commit:function(a,e){this.getValue()?e.setAttribute("cellSpacing",this.getValue()):e.removeAttribute("cellSpacing")}},{type:"text",id:"txtCellPad",requiredContent:"table[cellpadding]",controlStyle:"width:3em",label:a.lang.table.cellPad,"default":a.filter.check("table[cellpadding]")?
-1:0,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellPadding),setup:function(a){this.setValue(a.getAttribute("cellPadding")||"")},commit:function(a,e){this.getValue()?e.setAttribute("cellPadding",this.getValue()):e.removeAttribute("cellPadding")}}]}]},{type:"html",align:"right",html:""},{type:"vbox",padding:0,children:[{type:"text",id:"txtCaption",requiredContent:"caption",label:a.lang.table.caption,setup:function(a){this.enable();a=a.getElementsByTag("caption");if(0<a.count()){a=
+controlStyle:"width:3em",label:a.lang.table.cellSpace,"default":0,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellSpacing),setup:function(a){this.setValue(a.getAttribute("cellSpacing")||"")},commit:function(a,e){this.getValue()?e.setAttribute("cellSpacing",this.getValue()):e.removeAttribute("cellSpacing")}},{type:"text",id:"txtCellPad",requiredContent:"table[cellpadding]",controlStyle:"width:3em",label:a.lang.table.cellPad,"default":10,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellPadding),setup:function(a){this.setValue(a.getAttribute("cellPadding")||"")},commit:function(a,e){this.getValue()?e.setAttribute("cellPadding",this.getValue()):e.removeAttribute("cellPadding")}}]}]},{type:"html",align:"right",html:""},{type:"vbox",padding:0,children:[{type:"text",id:"txtCaption",requiredContent:"caption",label:a.lang.table.caption,setup:function(a){this.enable();a=a.getElementsByTag("caption");if(0<a.count()){a=
 a.getItem(0);var e=a.getFirst(CKEDITOR.dom.walker.nodeType(CKEDITOR.NODE_ELEMENT));e&&!e.equals(a.getBogus())?(this.disable(),this.setValue(a.getText())):(a=CKEDITOR.tools.trim(a.getText()),this.setValue(a))}},commit:function(d,e){if(this.isEnabled()){var b=this.getValue(),c=e.getElementsByTag("caption");if(b)0<c.count()?(c=c.getItem(0),c.setHtml("")):(c=new CKEDITOR.dom.element("caption",a.document),e.append(c,!0)),c.append(new CKEDITOR.dom.text(b,a.document));else if(0<c.count())for(b=c.count()-
 1;0<=b;b--)c.getItem(b).remove()}}},{type:"text",id:"txtSummary",bidi:!0,requiredContent:"table[summary]",label:a.lang.table.summary,setup:function(a){this.setValue(a.getAttribute("summary")||"")},commit:function(a,e){this.getValue()?e.setAttribute("summary",this.getValue()):e.removeAttribute("summary")}}]}]},q&&q.createAdvancedTab(a,null,"table")]}}var v=CKEDITOR.tools.cssLength,n=function(a){var f=this.id;a.info||(a.info={});a.info[f]=this.getValue()};CKEDITOR.dialog.add("table",function(a){return r(a,
-"table")});CKEDITOR.dialog.add("tableProperties",function(a){return r(a,"tableProperties")})})();
\ No newline at end of file
+"table")});CKEDITOR.dialog.add("tableProperties",function(a){return r(a,"tableProperties")})})();
diff -ur ./CRM/Contribute/xml/Menu/Contribute.xml /var/www/civicrm/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml
--- ./CRM/Contribute/xml/Menu/Contribute.xml	2020-08-06 12:54:52.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml	2020-08-31 21:12:08.113688289 +1000
@@ -223,6 +223,8 @@
     <title>Cancel Subscription</title>
     <page_callback>CRM_Contribute_Form_CancelSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/onbehalf</path>
@@ -240,6 +242,8 @@
     <title>Update Subscription</title>
     <page_callback>CRM_Contribute_Form_UpdateSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/subscriptionstatus</path>
diff -ur ./css/civicrm.css /var/www/civicrm/sites/all/modules/civicrm/css/civicrm.css
--- ./css/civicrm.css	2020-08-22 18:42:26.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/css/civicrm.css	2020-08-31 21:12:28.001891014 +1000
@@ -258,6 +258,14 @@
   height: 5px;
 }
 
+/*
+ * A fix to prevent the CBF theme making input fields 90% of the form width
+ */
+#crm-container form input.form-text,
+#crm-container form textarea.form-textarea {
+    width: auto;
+}
+
 .crm-container td.compressed select,
 .crm-container td.compressed input {
   font-size: 0.90em;
@@ -1503,6 +1511,7 @@
 }
 .crm-container a.action-item {
   display: inline;
+  padding: 2px;
 }
 
 /* theming for panel and context menus */
@@ -1809,6 +1818,10 @@
 
 /* Set/alter ICONS */
 
+#crm-container.crm-public div#printer-friendly {
+  display: none;
+}
+
 #crm-container .order-icon {
   height: 15px;
   width: 10px;
@@ -3134,6 +3147,9 @@
   text-overflow: ellipsis;
   font-weight: normal;
 }
+.crm-container .select2-results .select2-highlighted {
+  font-weight: bold;
+}
 .crm-container .select2-results .select2-highlighted > .select2-result-label .crm-select2-row-description p {
   color: #f0f0f0;
 }
@@ -3255,6 +3271,10 @@
 #crm-container .crm-fb-tweet-buttons {
   width: 93%;
 }
+#crm-container .crm-fb-tweet-buttons .label {
+  width:  64px;
+  height: 30px;
+}
 
 /* Checkbox gropus */
 .crm-container ul.crm-checkbox-list {
@@ -3660,6 +3680,7 @@
 #crm-container.crm-public .price-field-amount {
   padding-top: 6px;
   font-size: 15px;
+  line-height: 135%;
 }
 
 .crm-container.crm-public .select2-container,
@@ -3700,9 +3721,16 @@
   font-size: 15px;
 }
 
+#crm-container.crm-public .calc-value.description,
+#crm-container.crm-public .content.description {
+  padding-top: 0;
+  font-size: 0.9em;
+}
+
 #crm-container.crm-public .crm-section,
 .crm-section {
   margin-bottom: 0;
+  clear: left; /* Ensure form elements clear left */
 }
 
 #crm-container.crm-public #crm-submit-buttons {
diff -ur ./extern/url.php /var/www/civicrm/sites/all/modules/civicrm/extern/url.php
--- ./extern/url.php	2020-08-06 12:54:52.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/extern/url.php	2020-09-01 20:15:26.281560307 +1000
@@ -4,6 +4,8 @@
 require_once 'CRM/Core/Error.php';
 require_once 'CRM/Utils/Array.php';
 
+require_once '../../cbf/cbf_civicrm.module';
+
 CRM_Core_Config::singleton();
 
 // To keep backward compatibility for URLs generated
diff -ur ./js/Common.js /var/www/civicrm/sites/all/modules/civicrm/js/Common.js
--- ./js/Common.js	2020-08-22 18:42:26.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/js/Common.js	2020-08-31 21:12:33.657948694 +1000
@@ -1652,6 +1652,25 @@
     return (yiq >= 128) ? 'black' : 'white';
   };
 
+  // based on https://github.com/janl/mustache.js/blob/master/mustache.js
+  // If you feel the need to use this function, consider whether assembling HTML
+  // via DOM might be a cleaner approach rather than using string concatenation.
+  CRM.utils.escapeHtml = function(string) {
+    var entityMap = {
+      '&': '&amp;',
+      '<': '&lt;',
+      '>': '&gt;',
+      '"': '&quot;',
+      "'": '&#39;',
+      '/': '&#x2F;',
+      '`': '&#x60;',
+      '=': '&#x3D;'
+    };
+    return String(string).replace(/[&<>"'`=\/]/g, function fromEntityMap (s) {
+      return entityMap[s];
+    });
+  }
+
   // CVE-2015-9251 - Prevent auto-execution of scripts when no explicit dataType was provided
   $.ajaxPrefilter(function(s) {
     if (s.crossDomain) {
diff -ur ./js/wysiwyg/crm.ckeditor.js /var/www/civicrm/sites/all/modules/civicrm/js/wysiwyg/crm.ckeditor.js
--- ./js/wysiwyg/crm.ckeditor.js	2020-08-06 12:54:52.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/js/wysiwyg/crm.ckeditor.js	2020-08-31 21:12:33.661948735 +1000
@@ -64,12 +64,22 @@
       $(item).addClass('crm-wysiwyg-enabled');
 
       CKEDITOR.replace($(item)[0], {
-        filebrowserBrowseUrl: browseUrl + '&type=files',
-        filebrowserImageBrowseUrl: browseUrl + '&type=images',
-        filebrowserFlashBrowseUrl: browseUrl + '&type=flash',
-        filebrowserUploadUrl: uploadUrl + '&type=files',
-        filebrowserImageUploadUrl: uploadUrl + '&type=images',
-        filebrowserFlashUploadUrl: uploadUrl + '&type=flash',
+        filebrowserBrowseUrl      : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserImageBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserFlashBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserUploadUrl      : '',
+        filebrowserImageUploadUrl : '',
+        filebrowserFlashUploadUrl : '',
+        bodyClass                 : 'cke_editable',
+        contentsCss               : '/sites/all/themes/cbf2019/css/editor.css',
+        stylesSet                 : 'civicrm:/sites/all/themes/cbf2019/js/ckeditor.styles.js',
+        scayt_autoStartup         : true,
+        scayt_sLang               : 'en_GB',
+        forcePasteAsPlainText     : true,
+        // allowedContent: 'meta title style iframe ol ul li table tbody tr th td div h1 h2 h3 h4 h5 h6 p span a img i[*]{*}(*);br em strong center blockquote',
+        // disallowedContent: 'img{width,height,border*}',
+        // If it gets too hard, set allowedContent to true and remove disallowedContent.
+        allowedContent: true,
         customConfig: customConfig,
         on: {
           instanceReady: onReady
@@ -145,3 +155,20 @@
   };
 
 })(CRM.$, CRM._);
+
+/**
+ * IMCE support
+ * Copied from ckeditor.utils.js
+ */
+var ckeditor_imceSendTo = function (file, win){
+  var cfunc = win.location.href.split('&');
+
+  for (var x in cfunc) {
+    if (cfunc[x].match(/^CKEditorFuncNum=\d+$/)) {
+      cfunc = cfunc[x].split('=');
+      break;
+    }
+  }
+  CKEDITOR.tools.callFunction(cfunc[1], file.url);
+  win.close();
+};
diff -ur ./packages/HTML/QuickForm/Controller.php /var/www/civicrm/sites/all/modules/civicrm/packages/HTML/QuickForm/Controller.php
--- ./packages/HTML/QuickForm/Controller.php	2019-08-30 17:45:35.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/packages/HTML/QuickForm/Controller.php	2020-08-31 21:12:33.661948735 +1000
@@ -214,7 +214,7 @@
                 return $this->_actions[$actionName]->perform($page, $actionName);
                 break;
             default:
-                return PEAR::raiseError('HTML_QuickForm_Controller: Unhandled action "' . $actionName . '" in page "' . $page->getAttribute('id') . '"');
+                return PEAR::raiseError('HTML_QuickForm_Controller: Unhandled action "' . htmlspecialchars($actionName) . '" in page "' . $page->getAttribute('id') . '"');
         } // switch
     }
 
diff -ur ./packages/HTML/QuickForm/Rule/Email.php /var/www/civicrm/sites/all/modules/civicrm/packages/HTML/QuickForm/Rule/Email.php
--- ./packages/HTML/QuickForm/Rule/Email.php	2020-01-02 13:36:09.000000000 +1100
+++ /var/www/civicrm/sites/all/modules/civicrm/packages/HTML/QuickForm/Rule/Email.php	2020-08-31 21:12:33.661948735 +1000
@@ -88,10 +88,12 @@
                 if (checkdnsrr($tokens[1], 'MX') || checkdnsrr($tokens[1], 'A')) {
                     return true;
                 }
+                CRM_Core_Error::debug_log_message("Email address '$email' failed to pass the DNS check");
                 return false;
             }
             return true;
         }
+        CRM_Core_Error::debug_log_message("Email address '$email' failed to match the regex pattern");
         return false;
     } // end func validate
 
diff -ur ./packages/Smarty/internals/core.create_dir_structure.php /var/www/civicrm/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php
--- ./packages/Smarty/internals/core.create_dir_structure.php	2019-08-30 17:45:35.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php	2020-08-31 21:12:33.665948776 +1000
@@ -65,7 +65,7 @@
                 $_make_new_dir = true;
             }
 
-            if ($_make_new_dir && !file_exists($_new_dir) && !@mkdir($_new_dir, $smarty->_dir_perms) && !is_dir($_new_dir)) {
+            if ($_make_new_dir && !file_exists($_new_dir) && (!@mkdir($_new_dir) || !@chmod($_new_dir, 0770)) && !is_dir($_new_dir)) {
                 $smarty->trigger_error("problem creating directory '" . $_new_dir . "'");
                 return false;
             }
