diff -ur ./CRM/Contribute/xml/Menu/Contribute.xml /var/www/civicrm/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml
--- ./CRM/Contribute/xml/Menu/Contribute.xml	2024-02-09 13:36:10.000000000 +1100
+++ /var/www/civicrm/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml	2024-03-11 21:06:02.164472712 +1100
@@ -230,6 +230,8 @@
     <title>Cancel Subscription</title>
     <page_callback>CRM_Contribute_Form_CancelSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/onbehalf</path>
@@ -247,6 +249,8 @@
     <title>Update Subscription</title>
     <page_callback>CRM_Contribute_Form_UpdateSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/subscriptionstatus</path>
diff -ur ./css/civicrm.css /var/www/civicrm/sites/all/modules/civicrm/css/civicrm.css
--- ./css/civicrm.css	2024-03-09 12:41:26.000000000 +1100
+++ /var/www/civicrm/sites/all/modules/civicrm/css/civicrm.css	2024-03-11 21:08:04.752687676 +1100
@@ -161,6 +161,10 @@
   vertical-align: top;
 }
 
+.crm-container td.html-adjust {
+  word-break: break-word;
+}
+
 .crm-container .crm-row-child {
   margin: 0;
   padding: 3px 0 3px 0;
@@ -294,6 +298,14 @@
   height: 5px;
 }
 
+/*
+ * A fix to prevent the CBF theme making input fields 90% of the form width
+ */
+#crm-container form input.form-text,
+#crm-container form textarea.form-textarea {
+    width: auto;
+}
+
 .crm-container td.compressed select,
 .crm-container td.compressed input {
   font-size: 0.90em;
@@ -1546,6 +1558,7 @@
 }
 .crm-container a.action-item {
   display: inline;
+  padding: 2px;
 }
 
 /* theming for panel and context menus */
@@ -1851,6 +1864,10 @@
 
 /* Set/alter ICONS */
 
+#crm-container.crm-public div#printer-friendly {
+  display: none;
+}
+
 #crm-container .order-icon {
   height: 15px;
   width: 10px;
@@ -3119,6 +3136,9 @@
   text-overflow: ellipsis;
   font-weight: normal;
 }
+.crm-container .select2-results .select2-highlighted {
+  font-weight: bold;
+}
 .crm-container .select2-results .select2-highlighted > .select2-result-label .crm-select2-row-description p {
   color: #f0f0f0;
 }
@@ -3257,6 +3277,10 @@
 #crm-container .crm-fb-tweet-buttons {
   width: 93%;
 }
+#crm-container .crm-fb-tweet-buttons .label {
+  width:  64px;
+  height: 30px;
+}
 
 /* Checkbox gropus */
 .crm-container ul.crm-checkbox-list {
@@ -3667,6 +3691,7 @@
 #crm-container.crm-public .price-field-amount {
   padding-top: 6px;
   font-size: 15px;
+  line-height: 135%;
 }
 
 .crm-container.crm-public .select2-container,
@@ -3707,9 +3732,16 @@
   font-size: 15px;
 }
 
+#crm-container.crm-public .calc-value.description,
+#crm-container.crm-public .content.description {
+  padding-top: 0;
+  font-size: 0.9em;
+}
+
 #crm-container.crm-public .crm-section,
 .crm-section {
   margin-bottom: 0;
+  clear: left; /* Ensure form elements clear left */
 }
 
 #crm-container.crm-public #crm-submit-buttons {
diff -ur ./ext/ckeditor4/js/crm.ckeditor.js /var/www/civicrm/sites/all/modules/civicrm/ext/ckeditor4/js/crm.ckeditor.js
--- ./ext/ckeditor4/js/crm.ckeditor.js	2023-02-28 12:12:41.000000000 +1100
+++ /var/www/civicrm/sites/all/modules/civicrm/ext/ckeditor4/js/crm.ckeditor.js	2024-03-11 21:08:04.752687676 +1100
@@ -64,13 +64,21 @@
       $(item).addClass('crm-wysiwyg-enabled');
 
       CKEDITOR.replace($(item)[0], {
-        filebrowserBrowseUrl: browseUrl + '&type=files',
-        filebrowserImageBrowseUrl: browseUrl + '&type=images',
-        filebrowserFlashBrowseUrl: browseUrl + '&type=flash',
-        filebrowserUploadUrl: uploadUrl + '&type=files',
-        filebrowserImageUploadUrl: uploadUrl + '&type=images',
-        filebrowserFlashUploadUrl: uploadUrl + '&type=flash',
-        customConfig: customConfig,
+        filebrowserBrowseUrl      : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserImageBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserFlashBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserUploadUrl      : '',
+        filebrowserImageUploadUrl : '',
+        filebrowserFlashUploadUrl : '',
+        bodyClass                 : 'cke_editable',
+        contentsCss               : '/sites/all/themes/cbf2019/css/editor.css',
+        stylesSet                 : 'civicrm:/sites/all/themes/cbf2019/js/ckeditor.styles.js',
+        scayt_autoStartup         : true,
+        scayt_sLang               : 'en_GB',
+        forcePasteAsPlainText     : true,
+        disallowedContent         : 'img{width,height,border*}',
+        // allowedContent: 'meta title style iframe ol ul li table tbody tr th td div h1 h2 h3 h4 h5 h6 p span a img i[*]{*}(*);br em strong center blockquote',
+        customConfig              : customConfig,
         on: {
           instanceReady: onReady
         }
@@ -145,3 +153,20 @@
   };
 
 })(CRM.$, CRM._);
+
+/**
+ * IMCE support
+ * Copied from ckeditor.utils.js
+ */
+var ckeditor_imceSendTo = function (file, win){
+  var cfunc = win.location.href.split('&');
+
+  for (var x in cfunc) {
+    if (cfunc[x].match(/^CKEditorFuncNum=\d+$/)) {
+      cfunc = cfunc[x].split('=');
+      break;
+    }
+  }
+  CKEDITOR.tools.callFunction(cfunc[1], file.url);
+  win.close();
+};
diff -ur ./extern/url.php /var/www/civicrm/sites/all/modules/civicrm/extern/url.php
--- ./extern/url.php	2023-08-04 00:36:09.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/extern/url.php	2024-03-11 21:08:24.044747013 +1100
@@ -4,6 +4,8 @@
 require_once 'CRM/Core/Error.php';
 require_once 'CRM/Utils/Array.php';
 
+require_once '../../cbf/cbf_civicrm.module';
+
 CRM_Core_Config::singleton();
 
 // To keep backward compatibility for URLs generated
diff -ur ./packages/Smarty/internals/core.create_dir_structure.php /var/www/civicrm/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php
--- ./packages/Smarty/internals/core.create_dir_structure.php	2023-02-28 12:12:43.000000000 +1100
+++ /var/www/civicrm/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php	2024-03-11 21:08:24.044747013 +1100
@@ -65,7 +65,7 @@
                 $_make_new_dir = true;
             }
 
-            if ($_make_new_dir && !file_exists($_new_dir) && !@mkdir($_new_dir, $smarty->_dir_perms) && !is_dir($_new_dir)) {
+            if ($_make_new_dir && !file_exists($_new_dir) && (!@mkdir($_new_dir) || !@chmod($_new_dir, 0770)) && !is_dir($_new_dir)) {
                 $smarty->trigger_error("problem creating directory '" . $_new_dir . "'");
                 return false;
             }
diff -ur ./vendor/guzzlehttp/guzzle/src/Utils.php /var/www/civicrm/sites/all/modules/civicrm/vendor/guzzlehttp/guzzle/src/Utils.php
--- ./vendor/guzzlehttp/guzzle/src/Utils.php	2023-05-16 06:43:01.000000000 +1000
+++ /var/www/civicrm/sites/all/modules/civicrm/vendor/guzzlehttp/guzzle/src/Utils.php	2024-03-12 15:30:44.965036503 +1100
@@ -113,7 +113,7 @@
      */
     public static function defaultUserAgent(): string
     {
-        return sprintf('GuzzleHttp/%d', ClientInterface::MAJOR_VERSION);
+        return sprintf('GuzzleHttp/%d', 7); // ClientInterface::MAJOR_VERSION);
     }
 
     /**
