diff -ur ./ang/crmMailing/EditMailingCtrl/2step.html /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html
--- ./ang/crmMailing/EditMailingCtrl/2step.html	2015-05-20 16:30:37.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html	2015-05-29 11:46:22.850661299 +1000
@@ -21,9 +21,9 @@
             </div>
             <span ng-model="placeholder" crm-ui-validate="mailing.body_html || mailing.body_text"></span>
           </div>
-          <div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
+          <!-- div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
             <div crm-attachments="attachments"/>
-          </div>
+          </div -->
           <div crm-ui-tab id="tab-header" crm-title="ts('Header and Footer')">
             <div crm-mailing-block-header-footer crm-mailing="mailing"/>
           </div>
diff -ur ./ang/crmUi.js /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmUi.js
--- ./ang/crmUi.js	2015-05-20 16:30:37.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmUi.js	2015-05-29 11:46:22.858661445 +1000
@@ -397,12 +397,31 @@
             _.extend(ck.config, {
               width: '94%',
               height: '400',
-              filebrowserBrowseUrl: CRM.crmUi.browseUrl + '?cms=civicrm&type=files',
-              filebrowserImageBrowseUrl: CRM.crmUi.browseUrl + '?cms=civicrm&type=images',
-              filebrowserFlashBrowseUrl: CRM.crmUi.browseUrl + '?cms=civicrm&type=flash',
-              filebrowserUploadUrl: CRM.crmUi.uploadUrl + '?cms=civicrm&type=files',
-              filebrowserImageUploadUrl: CRM.crmUi.uploadUrl + '?cms=civicrm&type=images',
-              filebrowserFlashUploadUrl: CRM.crmUi.uploadUrl + '?cms=civicrm&type=flash',
+              filebrowserBrowseUrl      : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+              filebrowserImageBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+              filebrowserFlashBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+              filebrowserUploadUrl      : '',
+              filebrowserImageUploadUrl : '',
+              filebrowserFlashUploadUrl : '',
+              bodyClass                 : 'cke_editable',
+              contentsCss               : '/sites/all/themes/cbootf/css/editor.css',
+              scayt_autoStartup         : true,
+              scayt_sLang               : 'en_GB',
+              forcePasteAsPlainText     : true,
+              toolbar : [
+                 ['Source'],
+                 ['Cut','Copy','PasteText','-','SpellChecker', 'Scayt'],
+                 ['Undo','Redo','Find','Replace','-','SelectAll','RemoveFormat'],
+                 ['Image','Table','HorizontalRule','SpecialChar'],
+                 '/',
+                 ['Bold','Italic','Underline','Strike','-','Subscript','Superscript'],
+                 ['NumberedList','BulletedList','-','Outdent','Indent','Blockquote'],
+                 ['JustifyLeft','JustifyCenter','JustifyRight','JustifyBlock'],
+                 ['Link','Unlink','Anchor','Linkit','LinkToNode', 'LinkToMenu'],
+                 '/',
+                 ['TextColor','BGColor'],
+                 ['Maximize', 'ShowBlocks']
+              ],
             });
           }
 
diff -ur ./Civi/Angular/Page/Main.php /var/www/citybibleforum/sites/all/modules/civicrm/Civi/Angular/Page/Main.php
--- ./Civi/Angular/Page/Main.php	2015-05-20 16:30:37.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/Civi/Angular/Page/Main.php	2015-05-29 11:46:22.862661518 +1000
@@ -80,7 +80,8 @@
     $this->res->addScriptFile('civicrm', 'bower_components/angular/angular.min.js', 100, 'html-header', FALSE);
 
     // FIXME: crmUi depends on loading ckeditor, but ckeditor doesn't work with this aggregation.
-    $this->res->addScriptFile('civicrm', 'packages/ckeditor/ckeditor.js', 100, 'page-header', FALSE);
+    $this->res->addScriptFile('civicrm', '../ckeditor/includes/ckeditor.utils.js?', 100, 'page-header', FALSE);
+    $this->res->addScriptFile('civicrm', '../../libraries/ckeditor/ckeditor.js?', 100, 'page-header', FALSE);
 
     $headOffset = 0;
     $config = \CRM_Core_Config::singleton();
diff -ur ./CRM/Contribute/xml/Menu/Contribute.xml /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml
--- ./CRM/Contribute/xml/Menu/Contribute.xml	2015-05-20 16:30:37.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml	2015-05-29 11:46:22.874661736 +1000
@@ -229,6 +229,8 @@
     <title>Cancel Subscription</title>
     <page_callback>CRM_Contribute_Form_CancelSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/onbehalf</path>
@@ -246,6 +248,8 @@
     <title>Update Subscription</title>
     <page_callback>CRM_Contribute_Form_UpdateSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/subscriptionstatus</path>
diff -ur ./CRM/Mailing/Info.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/Info.php
--- ./CRM/Mailing/Info.php	2015-05-20 16:30:37.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/Info.php	2015-05-29 11:46:22.874661736 +1000
@@ -105,7 +105,7 @@
       'return' => array('id', 'name', 'scheduled_date'),
       'sequential' => 1,
       'options' => array(
-        'limit' => 500,
+        'limit' => 50,
         'sort' => 'is_archived asc, scheduled_date desc',
       ),
     ));
diff -ur ./css/civicrm.css /var/www/citybibleforum/sites/all/modules/civicrm/css/civicrm.css
--- ./css/civicrm.css	2015-05-20 16:30:37.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/css/civicrm.css	2015-05-29 11:46:22.886661955 +1000
@@ -241,6 +241,14 @@
   height: 5px;
 }
 
+/* 
+ * A fix to prevent the CBF theme making input fields 90% of the form width
+ */
+#crm-container form input.form-text, 
+#crm-container form textarea.form-textarea {
+    width: auto;
+}
+
 .crm-container td.compressed select,
 .crm-container td.compressed input {
   font-size: 0.90em;
@@ -1573,6 +1581,7 @@
 }
 .crm-container a.action-item {
   display: inline;
+  padding: 2px;
 }
 
 /* theming for panel and context menus */
@@ -3362,6 +3371,11 @@
 #crm-container .crm-fb-tweet-buttons {
   width: 93%;
 }
+#crm-container .crm-fb-tweet-buttons .label {
+  width:  64px;
+  height: 30px;
+}
+
 #crm-container div.finalconf-button {
   float: right;
   padding: 5px 0 0 0;
@@ -3788,6 +3802,7 @@
 #crm-container.crm-public .crm-section,
 .crm-section {
   margin-bottom: 0;
+  clear: left; /* Ensure form elements clear left */
 }
 
 #crm-container.crm-public #crm-submit-buttons {
diff -ur ./packages/HTML/QuickForm/ckeditor.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/ckeditor.php
--- ./packages/HTML/QuickForm/ckeditor.php	2015-05-20 16:30:54.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/ckeditor.php	2015-05-29 11:46:22.894662101 +1000
@@ -74,19 +74,54 @@
                         CKEDITOR.remove(CKEDITOR.instances['{$elementId}']);
                     }
                     if ( cj('#{$elementId}').val( ) == '' ) cj('#{$elementId}').val('&nbsp;');
-                    CKEDITOR.replace( '{$elementId}' );
+                    CKEDITOR.replace( '{$elementId}',
+                    {
+                       scayt_autoStartup :         true,
+                       scayt_sLang       :         'en_GB',
+                       forcePasteAsPlainText :     true,
+                       toolbar : [
+                          ['Source'],
+                          ['Cut','Copy','PasteText','-','SpellChecker', 'Scayt'],
+                          ['Undo','Redo','Find','Replace','-','SelectAll','RemoveFormat'],
+                          ['Image','Table','HorizontalRule','SpecialChar'],
+                          '/',
+                          ['Bold','Italic','Underline','Strike','-','Subscript','Superscript'],
+                          ['NumberedList','BulletedList','-','Outdent','Indent','Blockquote'],
+                          ['JustifyLeft','JustifyCenter','JustifyRight','JustifyBlock'],
+                          ['Link','Unlink','Anchor','Linkit','LinkToNode', 'LinkToMenu'],
+                          '/',
+                          ['TextColor','BGColor'],
+                          ['Maximize', 'ShowBlocks']
+                       ],
+                    } );
                     var editor = CKEDITOR.instances['{$elementId}'];
                     if ( editor ) {
                         editor.config.width              = '".$this->width."';
                         editor.config.height             = '".$this->height."';
-                        editor.config.filebrowserBrowseUrl      = '".$browseUrl."?cms=civicrm&type=files';
-                        editor.config.filebrowserImageBrowseUrl = '".$browseUrl."?cms=civicrm&type=images';
-                        editor.config.filebrowserFlashBrowseUrl = '".$browseUrl."?cms=civicrm&type=flash';
-                        editor.config.filebrowserUploadUrl      = '".$uploadUrl."?cms=civicrm&type=files';
-                        editor.config.filebrowserImageUploadUrl = '".$uploadUrl."?cms=civicrm&type=images';
-                        editor.config.filebrowserFlashUploadUrl = '".$uploadUrl."?cms=civicrm&type=flash';
+                        editor.config.filebrowserBrowseUrl      = '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo_{$elementId}|';
+                        editor.config.filebrowserImageBrowseUrl = '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo_{$elementId}|';
+                        editor.config.filebrowserFlashBrowseUrl = '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo_{$elementId}|';
+                        editor.config.filebrowserUploadUrl      = '';
+                        editor.config.filebrowserImageUploadUrl = '';
+                        editor.config.filebrowserFlashUploadUrl = '';
+                        editor.config.bodyClass                 = 'cke_editable';
+                        editor.config.contentsCss               = '/sites/all/themes/cbootf/css/editor.css';
                     }
                 });
+                /**
+                 * IMCE support
+                 */
+                var ckeditor_imceSendTo_{$elementId} = function (file, win){
+                    var cfunc = win.location.href.split('&');
+                    for (var x in cfunc) {
+                        if (cfunc[x].match(/^CKEditorFuncNum=\d+$/)) {
+                            cfunc = cfunc[x].split('=');
+                            break;
+                        }
+                    }
+                    CKEDITOR.tools.callFunction(cfunc[1], file.url);
+                    win.close();
+                }
             </script>";
             return $html;
         }
@@ -107,17 +142,38 @@
                       CKEDITOR.remove(CKEDITOR.instances['{$elementId}']);
                   }
                   if ( $('#{$elementId}').val( ) == '' ) $('#{$elementId}').val('&nbsp;');
-                  CKEDITOR.replace( '{$elementId}' );
+                  CKEDITOR.replace( '{$elementId}',
+                  {
+                     scayt_autoStartup :         true,
+                     scayt_sLang       :         'en_GB',
+                     forcePasteAsPlainText :     true,
+                     toolbar : [
+                        ['Source'],
+                        ['Cut','Copy','PasteText','-','SpellChecker', 'Scayt'],
+                        ['Undo','Redo','Find','Replace','-','SelectAll','RemoveFormat'],
+                        ['Image','Table','HorizontalRule','SpecialChar'],
+                        '/',
+                        ['Bold','Italic','Underline','Strike','-','Subscript','Superscript'],
+                        ['NumberedList','BulletedList','-','Outdent','Indent','Blockquote'],
+                        ['JustifyLeft','JustifyCenter','JustifyRight','JustifyBlock'],
+                        ['Link','Unlink','Anchor','Linkit','LinkToNode', 'LinkToMenu'],
+                        '/',
+                        ['TextColor','BGColor'],
+                        ['Maximize', 'ShowBlocks']
+                     ],
+                  } );
                   var editor = CKEDITOR.instances['{$elementId}'];
                   if ( editor ) {
                       editor.config.width              = '".$this->width."';
                       editor.config.height             = '".$this->height."';
-                      editor.config.filebrowserBrowseUrl      = '".$browseUrl."?cms=civicrm&type=files';
-                      editor.config.filebrowserImageBrowseUrl = '".$browseUrl."?cms=civicrm&type=images';
-                      editor.config.filebrowserFlashBrowseUrl = '".$browseUrl."?cms=civicrm&type=flash';
-                      editor.config.filebrowserUploadUrl      = '".$uploadUrl."?cms=civicrm&type=files';
-                      editor.config.filebrowserImageUploadUrl = '".$uploadUrl."?cms=civicrm&type=images';
-                      editor.config.filebrowserFlashUploadUrl = '".$uploadUrl."?cms=civicrm&type=flash';
+                      editor.config.filebrowserBrowseUrl      = '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo_{$elementId}|';
+                      editor.config.filebrowserImageBrowseUrl = '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo_{$elementId}|';
+                      editor.config.filebrowserFlashBrowseUrl = '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo_{$elementId}|';
+                      editor.config.filebrowserUploadUrl      = '';
+                      editor.config.filebrowserImageUploadUrl = '';
+                      editor.config.filebrowserFlashUploadUrl = '';
+                      editor.config.bodyClass                 = 'cke_editable';
+                      editor.config.contentsCss               = '/sites/all/themes/cbootf/css/editor.css';
                       editor.on( 'blur', function( e ) {
                         this.updateElement();
                         this.destroy(true);
@@ -130,7 +186,21 @@
                 $('#{$elementId}-plain').click(openWysiwyg);
                 $('#{$elementId}-plain').keypress(openWysiwyg);
               });
-          </script>";
+              /**
+               * IMCE support
+               */
+              var ckeditor_imceSendTo_{$elementId} = function (file, win){
+                  var cfunc = win.location.href.split('&');
+                  for (var x in cfunc) {
+                      if (cfunc[x].match(/^CKEditorFuncNum=\d+$/)) {
+                          cfunc = cfunc[x].split('=');
+                          break;
+                      }
+                  }
+                  CKEDITOR.tools.callFunction(cfunc[1], file.url);
+                  win.close();
+              }
+            </script>";
           return $html;
         }
     }
diff -ur ./packages/HTML/QuickForm/tinymce.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/tinymce.php
--- ./packages/HTML/QuickForm/tinymce.php	2015-05-20 16:30:54.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/tinymce.php	2015-05-29 11:46:22.898662174 +1000
@@ -160,7 +160,7 @@
         theme_advanced_buttons1_add: "separator,forecolor,backcolor,separator,link,unlink,separator,image,hr,emotions",
         theme_advanced_buttons2    : "separator,numlist,bullist,|,outdent,indent,cite,separator,justifyleft,justifycenter,justifyright",
         theme_advanced_buttons2_add: "justifyfull,separator,pastetext,pasteword,|,spellchecker,separator,removeformat,separator,|,undo,redo,|,code,|,fullscreen,help",
-        theme_advanced_buttons3    : "",
+        theme_advanced_buttons3    : "tablecontrols",
         theme_advanced_toolbar_location : "top",
         theme_advanced_toolbar_align : "left",
         theme_advanced_statusbar_location : "bottom",
diff -ur ./packages/Net/SMTP.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/Net/SMTP.php
--- ./packages/Net/SMTP.php	2015-05-20 16:30:54.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/Net/SMTP.php	2015-05-29 11:46:22.910662392 +1000
@@ -601,8 +601,6 @@
          * (SSL) socket connection. */
         if ($tls && version_compare(PHP_VERSION, '5.1.0', '>=') &&
             extension_loaded('openssl') && isset($this->_esmtp['STARTTLS']) &&
-            // CiviCRM: CRM-8744
-            ($this->_esmtp['STARTTLS'] == true) &&
             strncasecmp($this->host, 'ssl://', 6) !== 0) {
             /* Start the TLS connection attempt. */
             if (PEAR::isError($result = $this->_put('STARTTLS'))) {
diff -ur ./packages/Smarty/internals/core.create_dir_structure.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php
--- ./packages/Smarty/internals/core.create_dir_structure.php	2015-05-20 16:30:54.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php	2015-05-29 11:46:22.926662684 +1000
@@ -65,7 +65,7 @@
                 $_make_new_dir = true;
             }
 
-            if ($_make_new_dir && !file_exists($_new_dir) && !@mkdir($_new_dir, $smarty->_dir_perms) && !is_dir($_new_dir)) {
+            if ($_make_new_dir && !file_exists($_new_dir) && (!@mkdir($_new_dir) || !@chmod($_new_dir, 0770)) && !is_dir($_new_dir)) {
                 $smarty->trigger_error("problem creating directory '" . $_new_dir . "'");
                 return false;
             }
