diff -ur ./ang/crmMailing/EditMailingCtrl/2step.html /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html
--- ./ang/crmMailing/EditMailingCtrl/2step.html	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html	2016-04-16 22:10:17.441137617 +1000
@@ -21,9 +21,9 @@
             </div>
             <span ng-model="placeholder" crm-ui-validate="mailing.body_html || mailing.body_text"></span>
           </div>
-          <div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
+          <!-- div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
             <div crm-attachments="attachments"/>
-          </div>
+          </div -->
           <div crm-ui-tab id="tab-header" crm-title="ts('Header and Footer')">
             <div crm-mailing-block-header-footer crm-mailing="mailing"/>
           </div>
Only in /var/www/citybibleforum/sites/all/modules/civicrm/ang: crmUi.js.orig
Only in /var/www/citybibleforum/sites/all/modules/civicrm/ang: crmUi.js.rej
diff -ur ./api/v3/Generic/Setvalue.php /var/www/citybibleforum/sites/all/modules/civicrm/api/v3/Generic/Setvalue.php
--- ./api/v3/Generic/Setvalue.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/api/v3/Generic/Setvalue.php	2016-04-16 22:10:34.313451561 +1000
@@ -106,6 +106,7 @@
       break;
 
     case CRM_Utils_Type::T_DATE:
+    case CRM_Utils_Type::T_DATE + CRM_Utils_Type::T_TIME:
       $value = CRM_Utils_Type::escape($value, "Date", FALSE);
       if (!$value) {
         return civicrm_api3_create_error("Param '$field' is not a date. format YYYYMMDD or YYYYMMDDHHMMSS");
Only in /var/www/citybibleforum/sites/all/modules/civicrm/Civi/Angular/Page: Main.php.orig
Only in /var/www/citybibleforum/sites/all/modules/civicrm/Civi/Angular/Page: Main.php.rej
diff -ur ./CRM/Activity/BAO/Activity.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Activity/BAO/Activity.php
--- ./CRM/Activity/BAO/Activity.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Activity/BAO/Activity.php	2016-04-17 16:19:20.875829007 +1000
@@ -919,7 +919,7 @@
     foreach ($compInfo as $compObj) {
       if (!empty($compObj->info['showActivitiesInCore'])) {
         if ($compObj->info['name'] == 'CiviCampaign') {
-          $componentPermission = "administer {$compObj->name}";
+          $componentPermission = "interview campaign contacts";
         }
         else {
           $componentPermission = "access {$compObj->name}";
@@ -2338,7 +2338,8 @@
       'CiviMember' => array('access CiviMember'),
       'CiviReport' => array('access CiviReport'),
       'CiviContribute' => array('access CiviContribute'),
-      'CiviCampaign' => array('administer CiviCampaign'),
+      'CiviCampaign' => array('administer CiviCampaign',
+                               'interview campaign contacts'),
     );
 
     // Return early when it is case activity.
diff -ur ./CRM/Campaign/BAO/Campaign.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Campaign/BAO/Campaign.php
--- ./CRM/Campaign/BAO/Campaign.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Campaign/BAO/Campaign.php	2016-04-17 16:19:20.855828037 +1000
@@ -567,7 +567,8 @@
     if (!isset($allow)) {
       $allow = FALSE;
       if (CRM_Core_Permission::check('manage campaign') ||
-        CRM_Core_Permission::check('administer CiviCampaign')
+        CRM_Core_Permission::check('administer CiviCampaign') ||
+        CRM_Core_Permission::check('interview campaign contacts')
       ) {
         $allow = TRUE;
       }
diff -ur ./CRM/Campaign/BAO/Query.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Campaign/BAO/Query.php
--- ./CRM/Campaign/BAO/Query.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Campaign/BAO/Query.php	2016-04-17 16:19:20.851827843 +1000
@@ -362,7 +362,7 @@
       array('multiple' => 'multiple', 'class' => 'crm-select2')
     );
 
-    $showInterviewer = FALSE;
+    $showInterviewer = TRUE;
     if (CRM_Core_Permission::check('administer CiviCampaign')) {
       $showInterviewer = TRUE;
     }
diff -ur ./CRM/Campaign/Form/Petition/Signature.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Campaign/Form/Petition/Signature.php
--- ./CRM/Campaign/Form/Petition/Signature.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Campaign/Form/Petition/Signature.php	2016-04-17 16:19:20.847827649 +1000
@@ -299,7 +299,7 @@
     $this->addButtons(array(
         array(
           'type' => 'upload',
-          'name' => ts('Sign the Petition'),
+          'name' => ts('Sign up'),
           'isDefault' => TRUE,
         ),
       )
diff -ur ./CRM/Contribute/BAO/Contribution.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/BAO/Contribution.php
--- ./CRM/Contribute/BAO/Contribution.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/BAO/Contribution.php	2016-04-17 16:19:20.863828425 +1000
@@ -2400,6 +2400,10 @@
       $paymentObject = Civi\Payment\System::singleton()->getByProcessor($this->_relatedObjects['paymentProcessor']);
 
       $entityID = $entity = NULL;
+      if (isset($this->contribution_recur_id) && $this->contribution_recur_id != 0) {
+        $entity = 'recur';
+        $entityID = $this->contribution_recur_id;
+      }
       if (isset($ids['contribution'])) {
         $entity = 'contribution';
         $entityID = $ids['contribution'];
diff -ur ./CRM/Contribute/Form/AdditionalInfo.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/Form/AdditionalInfo.php
--- ./CRM/Contribute/Form/AdditionalInfo.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/Form/AdditionalInfo.php	2016-04-17 16:19:20.859828231 +1000
@@ -440,10 +440,27 @@
       //retrieve custom data
       $customGroup = array();
 
+      // Assume this receipt is being sent to an anonymous user. This code emulates
+      // CRM_Core_Permission::customGroup() being called for an anonymous user, and
+      // assumes that anonymous users are not granted the 'access all custom data'
+      // permission. Anonymous users are only shown public custom data groups.
+      $publicGroups = CRM_ACL_BAO_ACL::group(
+          CRM_Core_Permission::VIEW,
+          0,
+          'civicrm_custom_group',
+          CRM_Core_PseudoConstant::get(
+              'CRM_Core_DAO_CustomField',
+              'custom_group_id',
+              array('fresh' => FALSE)
+          )
+      );
+
       foreach ($form->_groupTree as $groupID => $group) {
         $customFields = $customValues = array();
         if ($groupID == 'info') {
           continue;
+        } elseif (in_array($groupID, $publicGroups) === FALSE) {
+          continue;
         }
         foreach ($group['fields'] as $k => $field) {
           $field['title'] = $field['label'];
diff -ur ./CRM/Contribute/Import/Parser/Contribution.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/Import/Parser/Contribution.php
--- ./CRM/Contribute/Import/Parser/Contribution.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/Import/Parser/Contribution.php	2016-04-17 16:19:20.867828619 +1000
@@ -256,6 +256,7 @@
     }
 
     $params = &$this->getActiveFieldParams();
+    $params['contact_type'] = 'Contribution';
     $formatted = array('version' => 3);
 
     // don't add to recent items, CRM-4399
diff -ur ./CRM/Contribute/xml/Menu/Contribute.xml /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml
--- ./CRM/Contribute/xml/Menu/Contribute.xml	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml	2016-04-16 22:11:05.562033237 +1000
@@ -229,6 +229,8 @@
     <title>Cancel Subscription</title>
     <page_callback>CRM_Contribute_Form_CancelSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/onbehalf</path>
@@ -246,6 +248,8 @@
     <title>Update Subscription</title>
     <page_callback>CRM_Contribute_Form_UpdateSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/subscriptionstatus</path>
Only in /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Core: Controller.php.orig
Only in /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Core: Controller.php.rej
diff -ur ./CRM/Core/Payment.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Core/Payment.php
--- ./CRM/Core/Payment.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Core/Payment.php	2016-04-17 16:19:20.879829201 +1000
@@ -709,7 +709,7 @@
         'maxlength' => 60,
         'autocomplete' => 'off',
       ),
-      'is_required' => TRUE,
+      'is_required' => FALSE,
     );
 
     $metadata['billing_middle_name'] = array(
@@ -735,7 +735,7 @@
         'maxlength' => 60,
         'autocomplete' => 'off',
       ),
-      'is_required' => TRUE,
+      'is_required' => FALSE,
     );
 
     $metadata["billing_street_address-{$billingLocationID}"] = array(
@@ -748,7 +748,7 @@
         'maxlength' => 60,
         'autocomplete' => 'off',
       ),
-      'is_required' => TRUE,
+      'is_required' => FALSE,
     );
 
     $metadata["billing_city-{$billingLocationID}"] = array(
@@ -761,7 +761,7 @@
         'maxlength' => 60,
         'autocomplete' => 'off',
       ),
-      'is_required' => TRUE,
+      'is_required' => FALSE,
     );
 
     $metadata["billing_state_province_id-{$billingLocationID}"] = array(
@@ -769,7 +769,7 @@
       'title' => ts('State/Province'),
       'name' => "billing_state_province_id-{$billingLocationID}",
       'cc_field' => TRUE,
-      'is_required' => TRUE,
+      'is_required' => FALSE,
     );
 
     $metadata["billing_postal_code-{$billingLocationID}"] = array(
@@ -782,7 +782,7 @@
         'maxlength' => 60,
         'autocomplete' => 'off',
       ),
-      'is_required' => TRUE,
+      'is_required' => FALSE,
     );
 
     $metadata["billing_country_id-{$billingLocationID}"] = array(
@@ -793,7 +793,7 @@
       'attributes' => array(
         '' => ts('- select -'),
       ) + CRM_Core_PseudoConstant::country(),
-      'is_required' => TRUE,
+      'is_required' => FALSE,
     );
     return $metadata;
   }
diff -ur ./CRM/Event/Form/Participant.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Event/Form/Participant.php
--- ./CRM/Event/Form/Participant.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Event/Form/Participant.php	2016-04-17 16:19:20.875829007 +1000
@@ -1597,9 +1597,28 @@
 
       $template = CRM_Core_Smarty::singleton();
       $customGroup = array();
+
+      // Assume the receipt is being sent to an anonymous user. This code emulates
+      // CRM_Core_Permission::customGroup() being called for an anonymous user, and
+      // assumes that anonymous users are not granted the 'access all custom data'
+      // permission. Anonymous users are only shown public custom data groups.
+      $publicGroups = CRM_ACL_BAO_ACL::group(
+        CRM_Core_Permission::VIEW,
+        0,
+        'civicrm_custom_group',
+        CRM_Core_PseudoConstant::get(
+            'CRM_Core_DAO_CustomField',
+            'custom_group_id',
+            array('fresh' => FALSE)
+        )
+      );
+
       //format submitted data
       foreach ($params['custom'] as $fieldID => $values) {
         foreach ($values as $fieldValue) {
+          if (in_array($fieldValue['custom_group_id'], $publicGroups) === FALSE) {
+            continue;
+          }
           $customFields[$fieldID]['id'] = $fieldID;
           $formattedValue = CRM_Core_BAO_CustomField::displayValue($fieldValue['value'], $fieldID, $participants[0]->id);
           $customGroup[$customFields[$fieldID]['groupTitle']][$customFields[$fieldID]['label']] = str_replace('&nbsp;', '', $formattedValue);
diff -ur ./CRM/Event/Form/Registration.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Event/Form/Registration.php
--- ./CRM/Event/Form/Registration.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Event/Form/Registration.php	2016-04-17 16:19:20.871828813 +1000
@@ -120,7 +120,7 @@
    *
    * @var array
    */
-  protected $_params;
+  public $_params;
 
   /**
    * The fields involved in this contribution page.
Only in /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/BAO: Mailing.php.orig
Only in /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/BAO: Mailing.php.rej
diff -ur ./CRM/Mailing/Info.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/Info.php
--- ./CRM/Mailing/Info.php	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/Info.php	2016-04-16 22:11:24.950394291 +1000
@@ -121,7 +121,7 @@
       'return' => array('id', 'name', 'scheduled_date'),
       'sequential' => 1,
       'options' => array(
-        'limit' => 500,
+        'limit' => 50,
         'sort' => 'is_archived asc, scheduled_date desc',
       ),
     ));
Only in /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing: Info.php.orig
diff -ur ./css/civicrm.css /var/www/citybibleforum/sites/all/modules/civicrm/css/civicrm.css
--- ./css/civicrm.css	2016-04-08 06:29:23.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/css/civicrm.css	2016-04-16 22:11:52.202913057 +1000
@@ -241,6 +241,14 @@
   height: 5px;
 }
 
+/* 
+ * A fix to prevent the CBF theme making input fields 90% of the form width
+ */
+#crm-container form input.form-text, 
+#crm-container form textarea.form-textarea {
+    width: auto;
+}
+
 .crm-container td.compressed select,
 .crm-container td.compressed input {
   font-size: 0.90em;
@@ -1481,6 +1489,7 @@
 }
 .crm-container a.action-item {
   display: inline;
+  padding: 2px;
 }
 
 /* theming for panel and context menus */
@@ -3569,6 +3578,7 @@
 #crm-container.crm-public .crm-section,
 .crm-section {
   margin-bottom: 0;
+  clear: left; /* Ensure form elements clear left */
 }
 
 #crm-container.crm-public #crm-submit-buttons {
Only in /var/www/citybibleforum/sites/all/modules/civicrm/css: civicrm.css.orig
Only in /var/www/citybibleforum/sites/all/modules/civicrm/css: civicrm.css.rej
diff -ur ./packages/HTML/QuickForm/Rule/Email.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/Rule/Email.php
--- ./packages/HTML/QuickForm/Rule/Email.php	2016-04-08 06:29:26.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/Rule/Email.php	2016-04-16 22:12:34.147902976 +1000
@@ -69,10 +69,12 @@
                 if (checkdnsrr($tokens[1], 'MX') || checkdnsrr($tokens[1], 'A')) {
                     return true;
                 }
+                CRM_Core_Error::debug_log_message("Email address $email failed to pass the DNS check");
                 return false;
             }
             return true;
         }
+        CRM_Core_Error::debug_log_message("Email address $email failed to match the regex pattern");
         return false;
     } // end func validate
 
Only in /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/Rule: Email.php.orig
diff -ur ./packages/Smarty/internals/core.create_dir_structure.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php
--- ./packages/Smarty/internals/core.create_dir_structure.php	2016-04-08 06:29:26.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php	2016-04-16 22:12:49.332260431 +1000
@@ -65,7 +65,7 @@
                 $_make_new_dir = true;
             }
 
-            if ($_make_new_dir && !file_exists($_new_dir) && !@mkdir($_new_dir, $smarty->_dir_perms) && !is_dir($_new_dir)) {
+            if ($_make_new_dir && !file_exists($_new_dir) && (!@mkdir($_new_dir) || !@chmod($_new_dir, 0770)) && !is_dir($_new_dir)) {
                 $smarty->trigger_error("problem creating directory '" . $_new_dir . "'");
                 return false;
             }
