diff -ur ./ang/crmMailing/BodyHtml.html /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/BodyHtml.html
--- ./ang/crmMailing/BodyHtml.html	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/BodyHtml.html	2016-10-11 16:20:13.850507035 +1100
@@ -14,7 +14,6 @@
         crm-ui-insert-rx="insert:body_html"
         ng-model="mailing.body_html"
         ng-blur="checkTokens(mailing, 'body_html', 'insert:body_html')"
-        class="crm-wysiwyg-fullpage"
         ></textarea>
       <span ng-model="body_html_tokens" crm-ui-validate="hasAllTokens(mailing, 'body_html')"></span>
       <div ng-show="htmlForm.$error.crmUiValidate" class="crmMailing-error-link">
diff -ur ./ang/crmMailing/EditMailingCtrl/2step.html /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html
--- ./ang/crmMailing/EditMailingCtrl/2step.html	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/ang/crmMailing/EditMailingCtrl/2step.html	2016-10-11 16:20:13.850507035 +1100
@@ -21,9 +21,9 @@
             </div>
             <span ng-model="placeholder" crm-ui-validate="mailing.body_html || mailing.body_text"></span>
           </div>
-          <div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
+          <!-- div crm-ui-tab id="tab-attachment" crm-title="ts('Attachments')">
             <div crm-attachments="attachments"/>
-          </div>
+          </div -->
           <div crm-ui-tab id="tab-header" crm-title="ts('Header and Footer')">
             <div crm-mailing-block-header-footer crm-mailing="mailing"/>
           </div>
diff -ur ./api/v3/Generic/Setvalue.php /var/www/citybibleforum/sites/all/modules/civicrm/api/v3/Generic/Setvalue.php
--- ./api/v3/Generic/Setvalue.php	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/api/v3/Generic/Setvalue.php	2016-10-11 16:20:13.850507035 +1100
@@ -106,6 +106,7 @@
       break;
 
     case CRM_Utils_Type::T_DATE:
+    case CRM_Utils_Type::T_DATE + CRM_Utils_Type::T_TIME:
       $value = CRM_Utils_Type::escape($value, "Date", FALSE);
       if (!$value) {
         return civicrm_api3_create_error("Param '$field' is not a date. format YYYYMMDD or YYYYMMDDHHMMSS");
diff -ur ./bower_components/ckeditor/plugins/table/dialogs/table.js /var/www/citybibleforum/sites/all/modules/civicrm/bower_components/ckeditor/plugins/table/dialogs/table.js
--- ./bower_components/ckeditor/plugins/table/dialogs/table.js	2016-09-08 05:57:59.000000000 +1000
+++ /var/www/citybibleforum/sites/all/modules/civicrm/bower_components/ckeditor/plugins/table/dialogs/table.js	2016-10-11 16:20:13.854507114 +1100
@@ -10,12 +10,12 @@
 f))for(h=0;h<b.$.rows.length;h++)g=new CKEDITOR.dom.element(b.$.rows[h].cells[0]),g.renameNode("th"),g.setAttribute("scope","row");if(this.hasColumnHeaders&&"col"!=f&&"both"!=f)for(k=0;k<b.$.rows.length;k++)h=new CKEDITOR.dom.element(b.$.rows[k]),"tbody"==h.getParent().getName()&&(g=new CKEDITOR.dom.element(h.$.cells[0]),g.renameNode("td"),g.removeAttribute("scope"));c.txtHeight?b.setStyle("height",c.txtHeight):b.removeStyle("height");c.txtWidth?b.setStyle("width",c.txtWidth):b.removeStyle("width");
 b.getAttribute("style")||b.removeAttribute("style")}if(this._.selectedElement)try{e.selectBookmarks(d)}catch(p){}else a.insertElement(b),setTimeout(function(){var e=new CKEDITOR.dom.element(b.$.rows[0].cells[0]),c=a.createRange();c.moveToPosition(e,CKEDITOR.POSITION_AFTER_START);c.select()},0)},contents:[{id:"info",label:a.lang.table.title,elements:[{type:"hbox",widths:[null,null],styles:["vertical-align:top"],children:[{type:"vbox",padding:0,children:[{type:"text",id:"txtRows","default":3,label:a.lang.table.rows,
 required:!0,controlStyle:"width:5em",validate:r(a.lang.table.invalidRows),setup:function(e){this.setValue(e.$.rows.length)},commit:l},{type:"text",id:"txtCols","default":2,label:a.lang.table.columns,required:!0,controlStyle:"width:5em",validate:r(a.lang.table.invalidCols),setup:function(e){this.setValue(v(e))},commit:l},{type:"html",html:"\x26nbsp;"},{type:"select",id:"selHeaders",requiredContent:"th","default":"",label:a.lang.table.headers,items:[[a.lang.table.headersNone,""],[a.lang.table.headersRow,
-"row"],[a.lang.table.headersColumn,"col"],[a.lang.table.headersBoth,"both"]],setup:function(e){var a=this.getDialog();a.hasColumnHeaders=!0;for(var b=0;b<e.$.rows.length;b++){var c=e.$.rows[b].cells[0];if(c&&"th"!=c.nodeName.toLowerCase()){a.hasColumnHeaders=!1;break}}null!==e.$.tHead?this.setValue(a.hasColumnHeaders?"both":"row"):this.setValue(a.hasColumnHeaders?"col":"")},commit:l},{type:"text",id:"txtBorder",requiredContent:"table[border]","default":a.filter.check("table[border]")?1:0,label:a.lang.table.border,
+"row"],[a.lang.table.headersColumn,"col"],[a.lang.table.headersBoth,"both"]],setup:function(e){var a=this.getDialog();a.hasColumnHeaders=!0;for(var b=0;b<e.$.rows.length;b++){var c=e.$.rows[b].cells[0];if(c&&"th"!=c.nodeName.toLowerCase()){a.hasColumnHeaders=!1;break}}null!==e.$.tHead?this.setValue(a.hasColumnHeaders?"both":"row"):this.setValue(a.hasColumnHeaders?"col":"")},commit:l},{type:"text",id:"txtBorder",requiredContent:"table[border]","default":0,label:a.lang.table.border,
 controlStyle:"width:3em",validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidBorder),setup:function(a){this.setValue(a.getAttribute("border")||"")},commit:function(a,d){this.getValue()?d.setAttribute("border",this.getValue()):d.removeAttribute("border")}},{id:"cmbAlign",type:"select",requiredContent:"table[align]","default":"",label:a.lang.common.align,items:[[a.lang.common.notSet,""],[a.lang.common.alignLeft,"left"],[a.lang.common.alignCenter,"center"],[a.lang.common.alignRight,"right"]],
-setup:function(a){this.setValue(a.getAttribute("align")||"")},commit:function(a,d){this.getValue()?d.setAttribute("align",this.getValue()):d.removeAttribute("align")}}]},{type:"vbox",padding:0,children:[{type:"hbox",widths:["5em"],children:[{type:"text",id:"txtWidth",requiredContent:"table{width}",controlStyle:"width:5em",label:a.lang.common.width,title:a.lang.common.cssLengthTooltip,"default":a.filter.check("table{width}")?500>q.getSize("width")?"100%":500:0,getValue:u,validate:CKEDITOR.dialog.validate.cssLength(a.lang.common.invalidCssLength.replace("%1",
+setup:function(a){this.setValue(a.getAttribute("align")||"")},commit:function(a,d){this.getValue()?d.setAttribute("align",this.getValue()):d.removeAttribute("align")}}]},{type:"vbox",padding:0,children:[{type:"hbox",widths:["5em"],children:[{type:"text",id:"txtWidth",requiredContent:"table{width}",controlStyle:"width:5em",label:a.lang.common.width,title:a.lang.common.cssLengthTooltip,"default":0,getValue:u,validate:CKEDITOR.dialog.validate.cssLength(a.lang.common.invalidCssLength.replace("%1",
 a.lang.common.width)),onChange:function(){var a=this.getDialog().getContentElement("advanced","advStyles");a&&a.updateStyle("width",this.getValue())},setup:function(a){a=a.getStyle("width");this.setValue(a)},commit:l}]},{type:"hbox",widths:["5em"],children:[{type:"text",id:"txtHeight",requiredContent:"table{height}",controlStyle:"width:5em",label:a.lang.common.height,title:a.lang.common.cssLengthTooltip,"default":"",getValue:u,validate:CKEDITOR.dialog.validate.cssLength(a.lang.common.invalidCssLength.replace("%1",
-a.lang.common.height)),onChange:function(){var a=this.getDialog().getContentElement("advanced","advStyles");a&&a.updateStyle("height",this.getValue())},setup:function(a){(a=a.getStyle("height"))&&this.setValue(a)},commit:l}]},{type:"html",html:"\x26nbsp;"},{type:"text",id:"txtCellSpace",requiredContent:"table[cellspacing]",controlStyle:"width:3em",label:a.lang.table.cellSpace,"default":a.filter.check("table[cellspacing]")?1:0,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellSpacing),
-setup:function(a){this.setValue(a.getAttribute("cellSpacing")||"")},commit:function(a,d){this.getValue()?d.setAttribute("cellSpacing",this.getValue()):d.removeAttribute("cellSpacing")}},{type:"text",id:"txtCellPad",requiredContent:"table[cellpadding]",controlStyle:"width:3em",label:a.lang.table.cellPad,"default":a.filter.check("table[cellpadding]")?1:0,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellPadding),setup:function(a){this.setValue(a.getAttribute("cellPadding")||"")},commit:function(a,
+a.lang.common.height)),onChange:function(){var a=this.getDialog().getContentElement("advanced","advStyles");a&&a.updateStyle("height",this.getValue())},setup:function(a){(a=a.getStyle("height"))&&this.setValue(a)},commit:l}]},{type:"html",html:"\x26nbsp;"},{type:"text",id:"txtCellSpace",requiredContent:"table[cellspacing]",controlStyle:"width:3em",label:a.lang.table.cellSpace,"default":0,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellSpacing),
+setup:function(a){this.setValue(a.getAttribute("cellSpacing")||"")},commit:function(a,d){this.getValue()?d.setAttribute("cellSpacing",this.getValue()):d.removeAttribute("cellSpacing")}},{type:"text",id:"txtCellPad",requiredContent:"table[cellpadding]",controlStyle:"width:3em",label:a.lang.table.cellPad,"default":5,validate:CKEDITOR.dialog.validate.number(a.lang.table.invalidCellPadding),setup:function(a){this.setValue(a.getAttribute("cellPadding")||"")},commit:function(a,
 d){this.getValue()?d.setAttribute("cellPadding",this.getValue()):d.removeAttribute("cellPadding")}}]}]},{type:"html",align:"right",html:""},{type:"vbox",padding:0,children:[{type:"text",id:"txtCaption",requiredContent:"caption",label:a.lang.table.caption,setup:function(a){this.enable();a=a.getElementsByTag("caption");if(0<a.count()){a=a.getItem(0);var d=a.getFirst(CKEDITOR.dom.walker.nodeType(CKEDITOR.NODE_ELEMENT));d&&!d.equals(a.getBogus())?(this.disable(),this.setValue(a.getText())):(a=CKEDITOR.tools.trim(a.getText()),
 this.setValue(a))}},commit:function(e,d){if(this.isEnabled()){var b=this.getValue(),c=d.getElementsByTag("caption");if(b)0<c.count()?(c=c.getItem(0),c.setHtml("")):(c=new CKEDITOR.dom.element("caption",a.document),d.append(c,!0)),c.append(new CKEDITOR.dom.text(b,a.document));else if(0<c.count())for(b=c.count()-1;0<=b;b--)c.getItem(b).remove()}}},{type:"text",id:"txtSummary",bidi:!0,requiredContent:"table[summary]",label:a.lang.table.summary,setup:function(a){this.setValue(a.getAttribute("summary")||
-"")},commit:function(a,d){this.getValue()?d.setAttribute("summary",this.getValue()):d.removeAttribute("summary")}}]}]},p&&p.createAdvancedTab(a,null,"table")]}}var u=CKEDITOR.tools.cssLength,l=function(a){var f=this.id;a.info||(a.info={});a.info[f]=this.getValue()};CKEDITOR.dialog.add("table",function(a){return q(a,"table")});CKEDITOR.dialog.add("tableProperties",function(a){return q(a,"tableProperties")})})();
\ No newline at end of file
+"")},commit:function(a,d){this.getValue()?d.setAttribute("summary",this.getValue()):d.removeAttribute("summary")}}]}]},p&&p.createAdvancedTab(a,null,"table")]}}var u=CKEDITOR.tools.cssLength,l=function(a){var f=this.id;a.info||(a.info={});a.info[f]=this.getValue()};CKEDITOR.dialog.add("table",function(a){return q(a,"table")});CKEDITOR.dialog.add("tableProperties",function(a){return q(a,"tableProperties")})})();
diff -ur ./CRM/Contact/Page/AJAX.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contact/Page/AJAX.php
--- ./CRM/Contact/Page/AJAX.php	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contact/Page/AJAX.php	2016-10-11 16:20:13.854507114 +1100
@@ -258,7 +258,7 @@
       ));
       // Save activity only for the primary (first) client
       if ($i == 0 && empty($result['is_error'])) {
-        CRM_Case_BAO_Case::createCaseRoleActivity($caseID, $result['id'], $relContactID);
+        CRM_Case_BAO_Case::createCaseRoleActivity($caseID, $result['id'], $relContactID, $sourceContactID);
       }
     }
 
diff -ur ./CRM/Contribute/xml/Menu/Contribute.xml /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml
--- ./CRM/Contribute/xml/Menu/Contribute.xml	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Contribute/xml/Menu/Contribute.xml	2016-10-11 16:20:13.854507114 +1100
@@ -229,6 +229,8 @@
     <title>Cancel Subscription</title>
     <page_callback>CRM_Contribute_Form_CancelSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/onbehalf</path>
@@ -246,6 +248,8 @@
     <title>Update Subscription</title>
     <page_callback>CRM_Contribute_Form_UpdateSubscription</page_callback>
     <access_arguments>make online contributions</access_arguments>
+    <is_ssl>true</is_ssl>
+    <is_public>true</is_public>
   </item>
   <item>
     <path>civicrm/contribute/subscriptionstatus</path>
diff -ur ./CRM/Core/xml/Menu/Admin.xml /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Core/xml/Menu/Admin.xml
--- ./CRM/Core/xml/Menu/Admin.xml	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Core/xml/Menu/Admin.xml	2016-10-11 16:20:13.854507114 +1100
@@ -741,7 +741,7 @@
      <path>civicrm/sms/send</path>
      <title>New Mass SMS</title>
      <page_callback>CRM_SMS_Controller_Send</page_callback>
-     <access_arguments>administer CiviCRM</access_arguments>
+     <access_arguments>access CiviMail</access_arguments>
      <page_type>1</page_type>
      <weight>610</weight>
   </item>
diff -ur ./CRM/Mailing/Info.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/Info.php
--- ./CRM/Mailing/Info.php	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Mailing/Info.php	2016-10-11 16:20:13.854507114 +1100
@@ -121,7 +121,7 @@
       'return' => array('id', 'name', 'scheduled_date'),
       'sequential' => 1,
       'options' => array(
-        'limit' => 500,
+        'limit' => 50,
         'sort' => 'is_archived asc, scheduled_date desc',
       ),
     ));
diff -ur ./CRM/Utils/DeprecatedUtils.php /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Utils/DeprecatedUtils.php
--- ./CRM/Utils/DeprecatedUtils.php	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/CRM/Utils/DeprecatedUtils.php	2016-10-11 16:20:13.858507193 +1100
@@ -450,7 +450,7 @@
               }
 
               if ($errorMsg) {
-                return civicrm_api3_create_error($errorMsg, $value[$key]);
+                return civicrm_api3_create_error($errorMsg);
               }
 
               // finally get soft credit contact id.
@@ -523,7 +523,7 @@
             }
           }
           else {
-            return civicrm_api3_create_error('No match found for specified contact in contribution data. Row was skipped.', 'pledge_payment');
+            return civicrm_api3_create_error('Contribution not found. Row was skipped.');
           }
         }
         else {
@@ -539,7 +539,7 @@
               $contributionContactID = $params['contribution_contact_id'] = $values['contribution_contact_id'] = $contact->id;
             }
             else {
-              return civicrm_api3_create_error('No match found for specified contact in contribution data. Row was skipped.', 'pledge_payment');
+              return civicrm_api3_create_error('Contact not found. Row was skipped.');
             }
           }
           else {
@@ -551,21 +551,21 @@
 
               // check if only one contact is found
               if (count($matchedIDs) > 1) {
-                return civicrm_api3_create_error($error['error_message']['message'], 'pledge_payment');
+                return civicrm_api3_create_error($error['error_message']['message']);
               }
               else {
                 $contributionContactID = $params['contribution_contact_id'] = $values['contribution_contact_id'] = $matchedIDs[0];
               }
             }
             else {
-              return civicrm_api3_create_error('No match found for specified contact in contribution data. Row was skipped.', 'pledge_payment');
+              return civicrm_api3_create_error('No match found for specified contact in contribution data. Row was skipped.');
             }
           }
         }
 
         if (!empty($params['pledge_id'])) {
           if (CRM_Core_DAO::getFieldValue('CRM_Pledge_DAO_Pledge', $params['pledge_id'], 'contact_id') != $contributionContactID) {
-            return civicrm_api3_create_error('Invalid Pledge ID provided. Contribution row was skipped.', 'pledge_payment');
+            return civicrm_api3_create_error('Invalid Pledge ID provided. Contribution row was skipped.');
           }
           $values['pledge_id'] = $params['pledge_id'];
         }
@@ -575,10 +575,10 @@
           $pledgeDetails = CRM_Pledge_BAO_Pledge::getContactPledges($contributionContactID);
 
           if (empty($pledgeDetails)) {
-            return civicrm_api3_create_error('No open pledges found for this contact. Contribution row was skipped.', 'pledge_payment');
+            return civicrm_api3_create_error('No open pledges found for this contact. Contribution row was skipped.');
           }
           elseif (count($pledgeDetails) > 1) {
-            return civicrm_api3_create_error('This contact has more than one open pledge. Unable to determine which pledge to apply the contribution to. Contribution row was skipped.', 'pledge_payment');
+            return civicrm_api3_create_error('This contact has more than one open pledge. Unable to determine which pledge to apply the contribution to. Contribution row was skipped.');
           }
 
           // this mean we have only one pending / in progress pledge
@@ -593,7 +593,7 @@
           $values['pledge_payment_id'] = $pledgePaymentDetails['id'];
         }
         else {
-          return civicrm_api3_create_error('Contribution and Pledge Payment amount mismatch for this record. Contribution row was skipped.', 'pledge_payment');
+          return civicrm_api3_create_error('Contribution and Pledge Payment amount mismatch for this record. Contribution row was skipped.');
         }
         break;
 
diff -ur ./css/civicrm.css /var/www/citybibleforum/sites/all/modules/civicrm/css/civicrm.css
--- ./css/civicrm.css	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/css/civicrm.css	2016-10-11 16:20:13.858507193 +1100
@@ -241,6 +241,14 @@
   height: 5px;
 }
 
+/* 
+ * A fix to prevent the CBF theme making input fields 90% of the form width
+ */
+#crm-container form input.form-text, 
+#crm-container form textarea.form-textarea {
+    width: auto;
+}
+
 .crm-container td.compressed select,
 .crm-container td.compressed input {
   font-size: 0.90em;
@@ -1481,6 +1489,7 @@
 }
 .crm-container a.action-item {
   display: inline;
+  padding: 2px;
 }
 
 /* theming for panel and context menus */
@@ -1804,6 +1813,10 @@
   margin: 0;
 }
 
+#crm-container.crm-public div#printer-friendly {
+  display: none;
+}
+
 #crm-container .order-icon {
   height: 15px;
   width: 10px;
@@ -3176,6 +3189,10 @@
 #crm-container .crm-fb-tweet-buttons {
   width: 93%;
 }
+#crm-container .crm-fb-tweet-buttons .label {
+  width:  64px;
+  height: 30px;
+}
 
 /* classes related to batch entry operation */
 .crm-container span.batch-edit,
@@ -3270,7 +3287,7 @@
 
 /* in place edit  */
 .crm-container .crm-editable-enabled {
-  white-space: nowrap;
+  white-space: normal;
   padding-left: 2px;
   border: 2px dashed transparent;
 }
@@ -3538,6 +3555,7 @@
 #crm-container.crm-public .price-field-amount {
   padding-top: 6px;
   font-size: 15px;
+  line-height: 135%;
 }
 
 .crm-container.crm-public .select2-container,
@@ -3575,9 +3593,16 @@
   font-size: 15px;
 }
 
+#crm-container.crm-public .calc-value.description,
+#crm-container.crm-public .content.description {
+  padding-top: 0;
+  font-size: 0.9em;
+}
+
 #crm-container.crm-public .crm-section,
 .crm-section {
   margin-bottom: 0;
+  clear: left; /* Ensure form elements clear left */
 }
 
 #crm-container.crm-public #crm-submit-buttons {
diff -ur ./js/wysiwyg/crm.ckeditor.js /var/www/citybibleforum/sites/all/modules/civicrm/js/wysiwyg/crm.ckeditor.js
--- ./js/wysiwyg/crm.ckeditor.js	2016-10-05 16:20:36.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/js/wysiwyg/crm.ckeditor.js	2016-10-11 16:20:13.858507193 +1100
@@ -71,12 +71,18 @@
       var isFullPage = $(item).hasClass('crm-wysiwyg-fullpage');
 
       CKEDITOR.replace($(item)[0], {
-        filebrowserBrowseUrl: browseUrl + '&type=files',
-        filebrowserImageBrowseUrl: browseUrl + '&type=images',
-        filebrowserFlashBrowseUrl: browseUrl + '&type=flash',
-        filebrowserUploadUrl: uploadUrl + '&type=files',
-        filebrowserImageUploadUrl: uploadUrl + '&type=images',
-        filebrowserFlashUploadUrl: uploadUrl + '&type=flash',
+        filebrowserBrowseUrl      : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserImageBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserFlashBrowseUrl : '/imce?app=ckeditor|sendto%40ckeditor_imceSendTo|',
+        filebrowserUploadUrl      : '',
+        filebrowserImageUploadUrl : '',
+        filebrowserFlashUploadUrl : '',
+        bodyClass                 : 'cke_editable',
+        contentsCss               : '/sites/all/themes/cbootf/css/editor.css',
+        stylesSet                 : 'civicrm:/sites/all/themes/cbootf/js/ckeditor.styles.js',
+        scayt_autoStartup         : true,
+        scayt_sLang               : 'en_GB',
+        forcePasteAsPlainText     : true,
         allowedContent: true, // For CiviMail!
         fullPage: isFullPage,
         customConfig: CRM.config.CKEditorCustomConfig,
@@ -157,3 +163,20 @@
   };
 
 })(CRM.$, CRM._);
+
+/**
+ * IMCE support
+ * Copied from ckeditor.utils.js
+ */
+var ckeditor_imceSendTo = function (file, win){
+  var cfunc = win.location.href.split('&');
+
+  for (var x in cfunc) {
+    if (cfunc[x].match(/^CKEditorFuncNum=\d+$/)) {
+      cfunc = cfunc[x].split('=');
+      break;
+    }
+  }
+  CKEDITOR.tools.callFunction(cfunc[1], file.url);
+  win.close();
+};
diff -ur ./packages/HTML/QuickForm/Rule/Email.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/Rule/Email.php
--- ./packages/HTML/QuickForm/Rule/Email.php	2016-10-05 16:20:38.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/HTML/QuickForm/Rule/Email.php	2016-10-11 16:20:13.862507272 +1100
@@ -69,10 +69,12 @@
                 if (checkdnsrr($tokens[1], 'MX') || checkdnsrr($tokens[1], 'A')) {
                     return true;
                 }
+                CRM_Core_Error::debug_log_message("Email address $email failed to pass the DNS check");
                 return false;
             }
             return true;
         }
+        CRM_Core_Error::debug_log_message("Email address $email failed to match the regex pattern");
         return false;
     } // end func validate
 
diff -ur ./packages/Net/SMTP.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/Net/SMTP.php
--- ./packages/Net/SMTP.php	2016-10-05 16:20:38.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/Net/SMTP.php	2016-10-11 16:20:13.862507272 +1100
@@ -184,7 +184,7 @@
         $this->pipelining = $pipelining;
 
         $this->_socket = new Net_Socket();
-        $this->_socket_options = $socket_options;
+        $this->_socket_options = array_merge((array) $socket_options, array('ssl' => array('verify_peer_name' => false)));
         $this->_timeout = $timeout;
 
         /* Include the Auth_SASL package.  If the package is available, we 
diff -ur ./packages/Smarty/internals/core.create_dir_structure.php /var/www/citybibleforum/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php
--- ./packages/Smarty/internals/core.create_dir_structure.php	2016-10-05 16:20:38.000000000 +1100
+++ /var/www/citybibleforum/sites/all/modules/civicrm/packages/Smarty/internals/core.create_dir_structure.php	2016-10-11 16:20:13.862507272 +1100
@@ -65,7 +65,7 @@
                 $_make_new_dir = true;
             }
 
-            if ($_make_new_dir && !file_exists($_new_dir) && !@mkdir($_new_dir, $smarty->_dir_perms) && !is_dir($_new_dir)) {
+            if ($_make_new_dir && !file_exists($_new_dir) && (!@mkdir($_new_dir) || !@chmod($_new_dir, 0770)) && !is_dir($_new_dir)) {
                 $smarty->trigger_error("problem creating directory '" . $_new_dir . "'");
                 return false;
             }
