diff --git a/paragraphs.field_widget.inc b/paragraphs.field_widget.inc
index c110062..20db506 100644
--- a/paragraphs.field_widget.inc
+++ b/paragraphs.field_widget.inc
@@ -705,7 +705,8 @@ function paragraphs_field_widget_embed_validate($element, &$form_state, $complet
 
           // Copied from _form_validate().
           if (isset($elements['#needs_validation'])) {
-            $is_empty_multiple = (!count($elements['#value']));
+            $is_countable = is_array($elements['#value']) || $elements['#value'] instanceof Countable;
+            $is_empty_multiple = $is_countable && (!count($elements['#value']));
             $is_empty_string = (is_string($elements['#value']) && drupal_strlen(trim($elements['#value'])) == 0);
             $is_empty_value = ($elements['#value'] === 0);
             if ($is_empty_multiple || $is_empty_string || $is_empty_value) {
